{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Finding a Findable Dataset\n",
    "\n",
    "```{dropdown} About this interactive ![icons](../static/img/rocket.png) recipe\n",
    "- Author(s): [Stuart Chalk](https://orcid.org/0000-0002-0703-7776)\n",
    "- Topic(s): How and where to find a 'findable' chemical dataset\n",
    "- Format(s): Interactive Jupyter Notebook (Python)\n",
    "- Scenario(s): You are looking for research data to complement your compare with your own data\n",
    "- Skill(s): You should be familiar with\n",
    "    - [Application Programming Interfaces (APIs)](https://www.ibm.com/topics/api)\n",
    "    - [Working with a data model](https://doi.org/10.1515/pac-2021-3013)\n",
    "    - [Introductory JSON](https://www.youtube.com/watch?v=iiADhChRriM)\n",
    "- Learning outcomes: After completing this example you should understand:\n",
    "    - How to make a request to a website using the Python 'requests' functionality\n",
    "    - Retrieve data in JSON format and how to parse it (knowing the data model)\n",
    "    - How to store confidential data in a remote file\n",
    "    - How programmtically you can authenticate to an API (one of many ways)\n",
    "- Citation: 'Finding a Findable Dataset', Stuart Chalk, The IUPAC FAIR Chemistry Cookbook, Contributed: 2024-02-14 [https://w3id.org/ifcc/IFCC013](https://w3id.org/ifcc/IFCC013).\n",
    "- Reuse: This notebook is made available under a [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/) license.\n",
    "```"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Scenario\n",
    "Our group has a set of thermophysical data on over 8000 chemical substances.  We want to integrate into this dataset another physical property dataset so that we can do an analysis of the correlations of the thermophysical data with the chosen physical property of the substances (that are common to both sets).\n",
    "\n",
    "Criteria for picking the physical property dataset: high quality, trusted, large, available with an open license, so I can publish the results and make the derived dataset open.\n",
    "- **High quality means**: unambiguous identification of each chemical substance, enough contextual information (metadata) to make the values scientifically useful, i.e., at least the composition of the solvent, the temperature and for volatile substances the pressure.\n",
    "- **Trusted means**: the provenance chain is reported with the data, and it shows that the data comes from a reputable source(s) and any aggregation and/or processing is documented in enough detail that the community can understand how the dataset has been created/provided."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Step 1 - Searching PubChem for datasets\n",
    "Pubchem houses a lot of data about chemical substances, compounds and bioassays.  Over time external organizations have worked with PubChem to include data, in one of a couple of ways:\n",
    "- data that has been integrated into PubChem pages (e.g., [CCDC](https://pubchem.ncbi.nlm.nih.gov/source/941) -> [example](https://pubchem.ncbi.nlm.nih.gov/compound/241))\n",
    "- data that is not available in a PubChem page but is available via the data sources section of the site as 'annotations' (e.g. [RCSB PDB](https://pubchem.ncbi.nlm.nih.gov/source/15751) -> [Example](https://pubchem.ncbi.nlm.nih.gov/source/15751#data=Annotations))\n",
    "\n",
    "The data available is may not be structured and or clearly described, however if the source has a website with an API then you are likely to get better quality metadata from the linked site."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 1.1 - Load the Python functions"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [],
   "source": [
    "# as these are direct imports (they do not reference a Python package) they are built into Python\n",
    "import requests\n",
    "import json"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-11-08T14:23:46.199477Z",
     "start_time": "2023-11-08T14:23:46.118025Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 1.2 - Search for sources that have 'curation efforts'"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[\n",
      "    {\n",
      "        \"name\": \"Agency for Toxic Substances and Disease Registry (ATSDR)\",\n",
      "        \"url\": \"https://www.atsdr.cdc.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Alliance of Genome Resources\",\n",
      "        \"url\": \"https://www.alliancegenome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Athena Minerals\",\n",
      "        \"url\": \"https://athena.unige.ch/athena/mineral/mineral.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Barrie Walker, BARK Information Services\",\n",
      "        \"url\": \"https://uk.linkedin.com/in/barrie-walker-85b4a510\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BindingDB\",\n",
      "        \"url\": \"https://www.bindingdb.org/rwd/bind/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioCyc\",\n",
      "        \"url\": \"https://biocyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioGRID\",\n",
      "        \"url\": \"https://thebiogrid.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CAMEO Chemicals\",\n",
      "        \"url\": \"https://cameochemicals.noaa.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Catalogue of Life (COL)\",\n",
      "        \"url\": \"https://www.catalogueoflife.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CCSbase\",\n",
      "        \"url\": \"https://ccsbase.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Line Ontology (CLO)\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Ontology (CL)\",\n",
      "        \"url\": \"https://obophenotype.github.io/cell-ontology/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEBI\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chebi/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEMBL\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chembl/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemical Probes Portal\",\n",
      "        \"url\": \"https://www.chemicalprobes.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChemIDplus\",\n",
      "        \"url\": \"https://pubchem.ncbi.nlm.nih.gov/source/ChemIDplus\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemoproteomic Metabolic Pathway Resource, Scripps University\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chris Southan\",\n",
      "        \"url\": \"https://www.ed.ac.uk/medicine-vet-medicine\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Comparative Toxicogenomics Database (CTD)\",\n",
      "        \"url\": \"http://ctdbase.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"COVID-19 Disease Map\",\n",
      "        \"url\": \"https://covid.pages.uni.lu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Database of Interacting Proteins (DIP)\",\n",
      "        \"url\": \"https://dip.doe-mbi.ucla.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Gene Interaction database (DGIdb)\",\n",
      "        \"url\": \"http://www.dgidb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Induced Liver Injury Rank (DILIrank) Dataset\",\n",
      "        \"url\": \"https://www.fda.gov/science-research/liver-toxicity-knowledge-base-ltkb/drug-induced-liver-injury-rank-dilirank-dataset\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugBank\",\n",
      "        \"url\": \"https://www.drugbank.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugCentral\",\n",
      "        \"url\": \"http://drugcentral.org\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"E. coli Metabolome Database (ECMDB)\",\n",
      "        \"url\": \"https://ecmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Egon Willighagen (Dept. of Bioinformatics - BiGCaT, Maastricht University)\",\n",
      "        \"url\": \"https://chem-bla-ics.blogspot.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA Air Toxics\",\n",
      "        \"url\": \"https://www3.epa.gov/ttn/atw/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA DSSTox\",\n",
      "        \"url\": \"https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Global Substance Registration System (GSRS)\",\n",
      "        \"url\": \"https://www.fda.gov/industry/fda-data-standards-advisory-board/fdas-global-substance-registration-system\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Orange Book\",\n",
      "        \"url\": \"https://www.fda.gov/drugs/drug-approvals-and-databases/approved-drug-products-therapeutic-equivalence-evaluations-orange-book\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Pharm Classes\",\n",
      "        \"url\": \"https://www.fda.gov/industry/structured-product-labeling-resources/pharmacologic-class\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gene Curation Coalition (GenCC)\",\n",
      "        \"url\": \"https://thegencc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gene Ontology (GO)\",\n",
      "        \"url\": \"http://geneontology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gloriam Group: Computational Receptor Biology - Department of Drug Design and Pharmacology, University of Copenhagen\",\n",
      "        \"url\": \"https://drug.ku.dk/disciplines/peptides-and-proteins/gloriam-group/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlycoNAVI\",\n",
      "        \"url\": \"https://glyconavi.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlyGen\",\n",
      "        \"url\": \"https://www.glygen.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Haz-Map, Information on Hazardous Chemicals and Occupational Diseases\",\n",
      "        \"url\": \"https://haz-map.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Chemical Information System (HCIS), Safe Work Australia\",\n",
      "        \"url\": \"http://hcis.safeworkaustralia.gov.au/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Substances Data Bank (HSDB)\",\n",
      "        \"url\": \"https://www.nlm.nih.gov/toxnet/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"HUGO Gene Nomenclature Committee (HGNC)\",\n",
      "        \"url\": \"https://www.genenames.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Human Metabolome Database (HMDB)\",\n",
      "        \"url\": \"https://hmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ILO-WHO International Chemical Safety Cards (ICSCs)\",\n",
      "        \"url\": \"https://www.ilo.org/safework/info/publications/WCMS_113134/lang--en/index.htm\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"INOH\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPAC Periodic Table of the Elements and Isotopes (IPTEI)\",\n",
      "        \"url\": \"https://iupac.org/iptei/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPHAR/BPS Guide to PHARMACOLOGY\",\n",
      "        \"url\": \"https://www.guidetopharmacology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"KEGG\",\n",
      "        \"url\": \"https://www.genome.jp/kegg/kegg2.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"KNApSAcK Species-Metabolite Database\",\n",
      "        \"url\": \"http://www.knapsackfamily.com/KNApSAcK/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Kruve Lab, Ionization & Mass Spectrometry, Stockholm University\",\n",
      "        \"url\": \"https://kruvelab.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Lab and Research Safety, University of Minnesota\",\n",
      "        \"url\": \"https://dehs.umn.edu/department-environmental-health-safety/lab-research-safety\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LIPID MAPS\",\n",
      "        \"url\": \"https://lipidmaps.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LOTUS - the natural products occurrence database\",\n",
      "        \"url\": \"https://lotus.naturalproducts.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"MarkerDB\",\n",
      "        \"url\": \"https://markerdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Medical Subject Headings (MeSH)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/mesh\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging and Contrast Agent Database (MICAD)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK5330/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging Database (MOLI)\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NCI Thesaurus (NCIt)\",\n",
      "        \"url\": \"https://ncit.nci.nih.gov/ncitbrowser/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NEQUIM - Chemoinformatics Group\",\n",
      "        \"url\": \"http://nequim.qui.ufmg.br/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NextMove Software\",\n",
      "        \"url\": \"https://www.nextmovesoftware.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIAID ChemDB\",\n",
      "        \"url\": \"https://chemdb.niaid.nih.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIST Chemistry WebBook\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NITE-CMC\",\n",
      "        \"url\": \"https://www.nite.go.jp/chem/english/ghs/ghs_index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Online Mendelian Inheritance in Man (OMIM)\",\n",
      "        \"url\": \"https://omim.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Open Targets\",\n",
      "        \"url\": \"https://www.opentargets.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PANTHER\",\n",
      "        \"url\": \"http://www.pantherdb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PathBank\",\n",
      "        \"url\": \"https://pathbank.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pathway Interaction Database\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pfam\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/interpro/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PharmGKB\",\n",
      "        \"url\": \"https://www.pharmgkb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pharos\",\n",
      "        \"url\": \"https://pharos.nih.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pistoia Alliance DataFAIRy Bioassay Pilot\",\n",
      "        \"url\": \"https://www.pistoiaalliance.org/projects/current-projects/datafairy-project/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Plant Reactome\",\n",
      "        \"url\": \"https://plantreactome.gramene.org/index.php?lang=en\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PlantCyc\",\n",
      "        \"url\": \"https://www.plantcyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Protein Ontology\",\n",
      "        \"url\": \"https://proconsortium.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"RCSB Protein Data Bank (RCSB PDB)\",\n",
      "        \"url\": \"https://www.rcsb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Reactome\",\n",
      "        \"url\": \"https://reactome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Rhea - Annotated Reactions Database\",\n",
      "        \"url\": \"https://www.rhea-db.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"RRUFF Project\",\n",
      "        \"url\": \"https://rruff.info/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"SureChEMBL\",\n",
      "        \"url\": \"https://www.surechembl.org/search/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics Cellosaurus\",\n",
      "        \"url\": \"https://www.cellosaurus.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics ENZYME\",\n",
      "        \"url\": \"https://enzyme.expasy.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Symbol Nomenclature for Glycans (SNFG) Reference Collection\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/glycans/snfg.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Cambridge Structural Database\",\n",
      "        \"url\": \"https://www.ccdc.cam.ac.uk/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The National Institute for Occupational Safety and Health (NIOSH)\",\n",
      "        \"url\": \"https://www.cdc.gov/niosh/npg/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Natural Products Atlas\",\n",
      "        \"url\": \"https://www.npatlas.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The University of Alabama Libraries\",\n",
      "        \"url\": \"https://ir.ua.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Therapeutic Target Database (TTD)\",\n",
      "        \"url\": \"https://db.idrblab.net/ttd/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Toxin and Toxin Target Database (T3DB)\",\n",
      "        \"url\": \"http://www.t3db.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniCarbKB\",\n",
      "        \"url\": \"\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniProt\",\n",
      "        \"url\": \"https://www.uniprot.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikidata\",\n",
      "        \"url\": \"https://www.wikidata.org/wiki/Wikidata:Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikipedia\",\n",
      "        \"url\": \"https://en.wikipedia.org/wiki/Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"World Register of Marine Species (WoRMS)\",\n",
      "        \"url\": \"https://www.marinespecies.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Yeast Metabolome Database (YMDB)\",\n",
      "        \"url\": \"http://www.ymdb.ca/\"\n",
      "    }\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "# This URL is the metadata about the data sources in PubChem\n",
    "url = 'https://pubchem.ncbi.nlm.nih.gov/rest/pug/sourcetable/all/JSON/?response_type=display'\n",
    "response = requests.get(url)\n",
    "srcs = response.json()\n",
    "results = []\n",
    "search = 'Curation Efforts'  # i.e., a repository, or other type of data source (this is in index 8 of the data list for each source)\n",
    "rows = srcs['Table']['Row']\n",
    "for row in rows:\n",
    "    if row['Cell'][8].find(search) != -1:\n",
    "        hit = {}\n",
    "        hit.update({'name': row['Cell'][0]})\n",
    "        hit.update({'url': row['Cell'][9]})\n",
    "        results.append(hit)\n",
    "# when printed this is a scrollable list of many entries\n",
    "print(json.dumps(results, indent=4))"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-11-08T14:23:46.770952Z",
     "start_time": "2023-11-08T14:23:46.133492Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Step 2 - Searching FAIR Sharing for datasets\n",
    "FAIRSharing is a database of FAIR resources and per se a database of datasets, however you might find a repository here that\n",
    "has the kind of data you are looking for. The code below accesses the FAIR sharing API so search for 'chemistry' (or other term) related resources."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "*Note: To use the code below please go to [https://fairsharing.org/accounts/signup](https://fairsharing.org/accounts/signup?ref=iupacfaircookbook), create an account and then enter your username and password in the quotes for 'fs_user' and 'fs_pass' below.*"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.1 - Authentication to the FAIRSharing API"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'success': True, 'jwt': 'eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI0YjVkNzQyMi1hOTg3LTRlZWYtYjQyYi1hN2U3Yjc2MTM0ZTEiLCJzdWIiOiI4NTA5Iiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNjk5NDUzNDMwLCJleHAiOjE2OTk1Mzk4MzB9.d0r8PUqy9nqaLS2-qPG8-20EHN1LnPunKmrPIvh9p7U', 'username': 'ChemCookbook', 'id': 8509, 'role': 'user', 'profile_type': 'none', 'watched_records': [], 'is_curator': False, 'is_super_curator': False, 'third_party': False, 'expiry': 1699539830, 'message': 'Authentication successful'}\n"
     ]
    }
   ],
   "source": [
    "# see https://fairsharing.org/API_doc for instructions on how to search the API\n",
    "# user login\n",
    "fs_user = \"ChemCookbook\"\n",
    "fs_pass = \"ydt_wdh_MRD*qut5xvq\"\n",
    "url = 'https://api.fairsharing.org/users/sign_in'\n",
    "loghdrs = {'Accept': 'application/json','Content-Type': 'application/json'}\n",
    "login = {'user': {'login': fs_user, 'password': fs_pass}}\n",
    "response = requests.request(\"POST\", url, headers=loghdrs, data=json.dumps(login))\n",
    "data = response.json()\n",
    "print(data)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-11-08T14:23:50.242822Z",
     "start_time": "2023-11-08T14:23:46.772306Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.2 - Make the API request"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"id\": \"1443\",\n",
      "    \"type\": \"fairsharing_records\",\n",
      "    \"attributes\": {\n",
      "        \"created_at\": \"2021-06-29T14:47:39.000Z\",\n",
      "        \"updated_at\": \"2023-03-15T08:08:11.434Z\",\n",
      "        \"metadata\": {\n",
      "            \"doi\": \"10.25504/FAIRsharing.cb1adb\",\n",
      "            \"name\": \"Portable reduced-precision binary format for trajectories produced by GROMACS package.\",\n",
      "            \"status\": \"ready\",\n",
      "            \"contacts\": [\n",
      "                {\n",
      "                    \"contact_name\": \"Adam Hospital\",\n",
      "                    \"contact_email\": \"adam.hospital@irbbarcelona.org\",\n",
      "                    \"contact_orcid\": \"0000-0002-8291-8071\"\n",
      "                }\n",
      "            ],\n",
      "            \"homepage\": \"https://manual.gromacs.org/documentation/2021/reference-manual/file-formats.html#xtc\",\n",
      "            \"citations\": [],\n",
      "            \"identifier\": 1443,\n",
      "            \"description\": \"The XTC format is a portable binary format for trajectories produced by GROMACS package. It uses the External Data Representation (xdr) routines for writing and reading data which were created for the Unix Network File System (NFS). Trajectories are written in XTC files using a reduced precision (lossy) algorithm which works multiplying the coordinates by a scaling factor (typically 1000), so converting them to pm (GROMACS standard distance unit is nm). This allows an integer rounding of the values. Several other tricks are performed, such as making use of atom proximity information: atoms close in sequence are usually close in space (e.g. water molecules). That makes XTC format the most efficient in terms of disk usage, in most cases reducing by a factor of 2 the size of any other binary trajectory format.\",\n",
      "            \"abbreviation\": \"xtc\",\n",
      "            \"support_links\": [\n",
      "                {\n",
      "                    \"url\": \"https://manual.gromacs.org/documentation/2021/reference-manual/file-formats.html#xtc\",\n",
      "                    \"name\": \"GROMACS reference manual\",\n",
      "                    \"type\": \"Help documentation\"\n",
      "                }\n",
      "            ],\n",
      "            \"year_creation\": 1995,\n",
      "            \"associated_tools\": [\n",
      "                {\n",
      "                    \"url\": \"https://www.gromacs.org/\",\n",
      "                    \"name\": \"GROMACS 2021\"\n",
      "                }\n",
      "            ],\n",
      "            \"deprecation_reason\": null\n",
      "        },\n",
      "        \"legacy_ids\": [\n",
      "            \"bsg-001620\",\n",
      "            \"bsg-s001620\"\n",
      "        ],\n",
      "        \"name\": \"FAIRsharing record for: Portable reduced-precision binary format for trajectories produced by GROMACS package.\",\n",
      "        \"abbreviation\": \"xtc\",\n",
      "        \"url\": \"https://fairsharing.org/10.25504/FAIRsharing.cb1adb\",\n",
      "        \"doi\": \"10.25504/FAIRsharing.cb1adb\",\n",
      "        \"fairsharing_licence\": \"https://creativecommons.org/licenses/by-sa/4.0/. Please link to https://fairsharing.org and https://api.fairsharing.org/img/fairsharing-attribution.svg for attribution.\",\n",
      "        \"description\": \"This FAIRsharing record describes: The XTC format is a portable binary format for trajectories produced by GROMACS package. It uses the External Data Representation (xdr) routines for writing and reading data which were created for the Unix Network File System (NFS). Trajectories are written in XTC files using a reduced precision (lossy) algorithm which works multiplying the coordinates by a scaling factor (typically 1000), so converting them to pm (GROMACS standard distance unit is nm). This allows an integer rounding of the values. Several other tricks are performed, such as making use of atom proximity information: atoms close in sequence are usually close in space (e.g. water molecules). That makes XTC format the most efficient in terms of disk usage, in most cases reducing by a factor of 2 the size of any other binary trajectory format.\",\n",
      "        \"linked_records\": [],\n",
      "        \"linking_records\": [],\n",
      "        \"fairsharing_registry\": \"Standard\",\n",
      "        \"record_type\": \"model_and_format\",\n",
      "        \"subjects\": [\n",
      "            \"Computational Chemistry\"\n",
      "        ],\n",
      "        \"domains\": [],\n",
      "        \"taxonomies\": [\n",
      "            \"Not applicable\"\n",
      "        ],\n",
      "        \"user_defined_tags\": [],\n",
      "        \"countries\": [\n",
      "            \"Netherlands\",\n",
      "            \"Sweden\"\n",
      "        ],\n",
      "        \"publications\": [\n",
      "            {\n",
      "                \"id\": 2540,\n",
      "                \"pubmed_id\": null,\n",
      "                \"title\": \"Methods and Techniques in Computational Chemistry: METECC-95\",\n",
      "                \"year\": 1995,\n",
      "                \"url\": \"https://www.semanticscholar.org/paper/Methods-and-techniques-in-computational-chemistry-%3A-Clementi-Corongiu/40385d32372c7c8af5dd4fe1ee442a47bc8ad87d\",\n",
      "                \"authors\": \"D. Green, K. Meacham, M. Surridge, F. van Hoesel and J. Berendsen\",\n",
      "                \"journal\": \"Methods and Techniques in Computational Chemistry: METECC-95\",\n",
      "                \"doi\": null,\n",
      "                \"created_at\": \"2021-09-30T08:27:11.512Z\",\n",
      "                \"updated_at\": \"2021-09-30T08:27:11.512Z\"\n",
      "            }\n",
      "        ],\n",
      "        \"licence_links\": [],\n",
      "        \"grants\": [\n",
      "            {\n",
      "                \"id\": 7488,\n",
      "                \"fairsharing_record_id\": 1443,\n",
      "                \"organisation_id\": 230,\n",
      "                \"relation\": \"maintains\",\n",
      "                \"created_at\": \"2021-09-30T09:28:39.405Z\",\n",
      "                \"updated_at\": \"2021-09-30T09:28:39.405Z\",\n",
      "                \"grant_id\": null,\n",
      "                \"is_lead\": true,\n",
      "                \"saved_state\": {\n",
      "                    \"id\": 230,\n",
      "                    \"name\": \"BioExcel CoE\",\n",
      "                    \"types\": [\n",
      "                        \"Consortium\"\n",
      "                    ],\n",
      "                    \"is_lead\": true,\n",
      "                    \"relation\": \"maintains\"\n",
      "                }\n",
      "            },\n",
      "            {\n",
      "                \"id\": 7489,\n",
      "                \"fairsharing_record_id\": 1443,\n",
      "                \"organisation_id\": 913,\n",
      "                \"relation\": \"funds\",\n",
      "                \"created_at\": \"2021-09-30T09:28:39.443Z\",\n",
      "                \"updated_at\": \"2021-09-30T09:32:30.296Z\",\n",
      "                \"grant_id\": 1650,\n",
      "                \"is_lead\": false,\n",
      "                \"saved_state\": {\n",
      "                    \"id\": 913,\n",
      "                    \"name\": \"European Commision\",\n",
      "                    \"grant\": \"675728\",\n",
      "                    \"types\": [\n",
      "                        \"Undefined\"\n",
      "                    ],\n",
      "                    \"is_lead\": false,\n",
      "                    \"relation\": \"funds\"\n",
      "                }\n",
      "            }\n",
      "        ],\n",
      "        \"url_for_logo\": null,\n",
      "        \"exhaustive_licences\": false\n",
      "    }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# in order to authenticate when making an API request the 'jwt' code above must\n",
    "# be included in the JSON headers (see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields)\n",
    "jwt = data['jwt']\n",
    "srchdrs = {'Accept': 'application/json', 'Content-Type': 'application/json', 'Authorization': \"Bearer {0}\".format(jwt)}\n",
    "searchterm = 'chemistry'\n",
    "searchurl ='https://api.fairsharing.org/search/fairsharing_records?q=' + searchterm\n",
    "search = requests.request(\"POST\", searchurl, headers=srchdrs)\n",
    "hits = json.loads(search.content)\n",
    "# this prints out the raw JSON for the first entry (the 'data' entry is a JSON list)\n",
    "# that is returned from the API request (formatted nicely, which means its on many lines)\n",
    "print(json.dumps(hits['data'][0], indent=4))"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-11-08T14:23:55.297853Z",
     "start_time": "2023-11-08T14:23:50.245422Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.3 - Output the data in a presentable format"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FAIRsharing record for: Portable reduced-precision binary format for trajectories produced by GROMACS package.: https://fairsharing.org/10.25504/FAIRsharing.cb1adb\n",
      "FAIRsharing record for: Chemistry: https://fairsharing.org/fairsharing_records/3524\n",
      "FAIRsharing record for: Chemistry vocabulary: https://fairsharing.org/10.25504/FAIRsharing.TrcBD2\n",
      "FAIRsharing record for: EMODnet Chemistry: https://fairsharing.org/10.25504/FAIRsharing.KOiDmy\n",
      "FAIRsharing record for: ioChem-BD: https://fairsharing.org/10.25504/FAIRsharing.lwW6a1\n",
      "FAIRsharing record for: Royal Society of Chemistry - Data policy: https://fairsharing.org/10.25504/FAIRsharing.egbgwm\n",
      "FAIRsharing record for: MINAS - A Database of Metal Ions in Nucleic AcidS: https://fairsharing.org/10.25504/FAIRsharing.wqtfkv\n",
      "FAIRsharing record for: Beilstein Journal of Organic Chemistry: https://fairsharing.org/10.25504/FAIRsharing.7GA79k\n",
      "FAIRsharing record for: CAS Registry Number: https://fairsharing.org/10.25504/FAIRsharing.r7Kwy7\n",
      "FAIRsharing record for: ChemSpider: https://fairsharing.org/10.25504/FAIRsharing.96f3gm\n",
      "FAIRsharing record for: Swedish Ocean Archive: https://fairsharing.org/10.25504/FAIRsharing.npf403\n",
      "FAIRsharing record for: Chemical Markup Language: https://fairsharing.org/10.25504/FAIRsharing.3mdt9n\n",
      "FAIRsharing record for: NFDI4Chem : https://fairsharing.org/fairsharing_records/5027\n",
      "FAIRsharing record for: CHARMM Card File Format: https://fairsharing.org/10.25504/FAIRsharing.7hp91k\n",
      "FAIRsharing record for: IUPAC International Chemical Identifier: https://fairsharing.org/10.25504/FAIRsharing.ddk9t9\n",
      "FAIRsharing record for: Chemical Abstracts Service Common Chemistry: https://fairsharing.org/10.25504/FAIRsharing.e5c86e\n",
      "FAIRsharing record for: Data Standard for Sharing Quantitative Results in Mass Spectrometry Metabolomics: https://fairsharing.org/10.25504/FAIRsharing.207caf\n",
      "FAIRsharing record for: American Chemical Society - Journal of the American Chemical Society - Author Guidelines: https://fairsharing.org/10.25504/FAIRsharing.f2b982\n",
      "FAIRsharing record for: Dot Bracket Notation (DBN) - Vienna Format: https://fairsharing.org/10.25504/FAIRsharing.4xrzw1\n",
      "FAIRsharing record for: Name Reaction Ontology: https://fairsharing.org/10.25504/FAIRsharing.w4tncg\n",
      "FAIRsharing record for: Molecular Process Ontology: https://fairsharing.org/10.25504/FAIRsharing.mct09a\n",
      "FAIRsharing record for: Chemical Methods Ontology: https://fairsharing.org/10.25504/FAIRsharing.9j4wh2\n",
      "FAIRsharing record for: MolMeDB: Molecules on Membranes Database: https://fairsharing.org/10.25504/FAIRsharing.CWzk3C\n",
      "FAIRsharing record for: Geological and Environmental Reference Materials: https://fairsharing.org/10.25504/FAIRsharing.219aec\n",
      "FAIRsharing record for: IUPAC Compendium of Chemical Terminology: https://fairsharing.org/10.25504/FAIRsharing.OsiqlF\n"
     ]
    }
   ],
   "source": [
    "# here we loop over the data that has been returned and print it out, one per line\n",
    "for hit in hits['data']:\n",
    "    print(hit['attributes']['name'] + \": \" + hit['attributes']['url'])"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-11-08T14:23:55.303432Z",
     "start_time": "2023-11-08T14:23:55.299129Z"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
