{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Recipe: Reusing a Chemical Dataset. Part 1- Finding Datasets\n",
    "## Scenario\n",
    "Our group has a set of thermophysical data on over 8000 chemical substances.  We want to integrate into this dataset another physical property dataset so that we can do an analysis of the correlations of the thermophysical data with the chosen physical property of the substances (that are common to both sets).\n",
    "\n",
    "Criteria for picking the physical property dataset: high quality, trusted, large, available with an open license, so I can publish the results and make the derived dataset open.\n",
    "- **High quality means**: unambiguous identification of each chemical substance, enough contextual information (metadata) to make the values scientifically useful, i.e., at least the composition of the solvent, the temperature and for volatile substances the pressure.\n",
    "- **Trusted means**: the provenance chain is reported with the data, and it shows that the data comes from a reputable source(s) and any aggregation and/or processing is documented in enough detail that the community can understand how the dataset has been created/provided."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Searching PubChem for datasets\n",
    "Pubchem houses a lot of data about chemical substances, compounds and bioassays.  Over time external organizations have worked with\n",
    "PubChem to include data, in one of a couple of ways:\n",
    "- data that has been integrated into PubChem pages (e.g., [CCDC](https://pubchem.ncbi.nlm.nih.gov/source/941) -> [example]('https://pubchem.ncbi.nlm.nih.gov/compound/241))\n",
    "- data that is not available in a PubChem page but is available via the data sources section of the site as 'annotations' (e.g. [RCSB PDB](https://pubchem.ncbi.nlm.nih.gov/source/15751) -> [Example](https://pubchem.ncbi.nlm.nih.gov/source/15751#data=Annotations))\n",
    "\n",
    "The data available is may not be structured and or clearly described, however if the source has a website with an API then you are likely to get better quality metadata from the linked site."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[\n",
      "    {\n",
      "        \"name\": \"Alliance of Genome Resources\",\n",
      "        \"url\": \"https://www.alliancegenome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Barrie Walker, BARK Information Services\",\n",
      "        \"url\": \"https://uk.linkedin.com/in/barrie-walker-85b4a510\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BindingDB\",\n",
      "        \"url\": \"https://www.bindingdb.org/rwd/bind/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioCyc\",\n",
      "        \"url\": \"https://biocyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioGRID\",\n",
      "        \"url\": \"https://thebiogrid.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CAMEO Chemicals\",\n",
      "        \"url\": \"https://cameochemicals.noaa.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Catalogue of Life (COL)\",\n",
      "        \"url\": \"https://www.catalogueoflife.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CCSbase\",\n",
      "        \"url\": \"https://ccsbase.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CDC-ATSDR Toxic Substances Portal\",\n",
      "        \"url\": \"https://www.atsdr.cdc.gov/features/toxicsubstances/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Line Ontology (CLO)\",\n",
      "        \"url\": \"http://www.clo-ontology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Ontology (CL)\",\n",
      "        \"url\": \"http://www.obofoundry.org/ontology/cl.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEBI\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chebi/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEMBL\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chembl/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemical Probes Portal\",\n",
      "        \"url\": \"https://www.chemicalprobes.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChemIDplus\",\n",
      "        \"url\": \"https://pubchem.ncbi.nlm.nih.gov/source/ChemIDplus\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemoproteomic Metabolic Pathway Resource, Scripps University\",\n",
      "        \"url\": \"https://rc2resource.scripps.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chris Southan\",\n",
      "        \"url\": \"https://www.ed.ac.uk/medicine-vet-medicine\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Comparative Toxicogenomics Database (CTD)\",\n",
      "        \"url\": \"http://ctdbase.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"COVID-19 Disease Map\",\n",
      "        \"url\": \"https://covid.pages.uni.lu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Gene Interaction database (DGIdb)\",\n",
      "        \"url\": \"http://www.dgidb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Induced Liver Injury Rank (DILIrank) Dataset\",\n",
      "        \"url\": \"https://www.fda.gov/science-research/liver-toxicity-knowledge-base-ltkb/drug-induced-liver-injury-rank-dilirank-dataset\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugBank\",\n",
      "        \"url\": \"https://www.drugbank.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugCentral\",\n",
      "        \"url\": \"http://drugcentral.org\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"E. coli Metabolome Database (ECMDB)\",\n",
      "        \"url\": \"https://ecmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Egon Willighagen (Dept. of Bioinformatics - BiGCaT, Maastricht University)\",\n",
      "        \"url\": \"https://chem-bla-ics.blogspot.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA Air Toxics\",\n",
      "        \"url\": \"https://www3.epa.gov/ttn/atw/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA DSSTox\",\n",
      "        \"url\": \"https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Essentials of Glycobiology, 3rd edition\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK310273/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Orange Book\",\n",
      "        \"url\": \"https://www.fda.gov/drugs/drug-approvals-and-databases/approved-drug-products-therapeutic-equivalence-evaluations-orange-book\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Pharm Classes\",\n",
      "        \"url\": \"https://www.fda.gov/industry/structured-product-labeling-resources/pharmacologic-class\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gene Ontology (GO)\",\n",
      "        \"url\": \"http://geneontology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gloriam Group: Computational Receptor Biology - Department of Drug Design and Pharmacology, University of Copenhagen\",\n",
      "        \"url\": \"https://drug.ku.dk/disciplines/peptides-and-proteins/gloriam-group/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlycoNAVI\",\n",
      "        \"url\": \"https://glyconavi.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlyGen\",\n",
      "        \"url\": \"https://www.glygen.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Haz-Map, Information on Hazardous Chemicals and Occupational Diseases\",\n",
      "        \"url\": \"https://haz-map.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Chemical Information System (HCIS), Safe Work Australia\",\n",
      "        \"url\": \"http://hcis.safeworkaustralia.gov.au/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Substances Data Bank (HSDB)\",\n",
      "        \"url\": \"https://www.nlm.nih.gov/toxnet/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"HUGO Gene Nomenclature Committee (HGNC)\",\n",
      "        \"url\": \"https://www.genenames.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Human Metabolome Database (HMDB)\",\n",
      "        \"url\": \"https://hmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ILO International Chemical Safety Cards (ICSC)\",\n",
      "        \"url\": \"http://www.ilo.org/dyn/icsc/showcard.home\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"INOH\",\n",
      "        \"url\": \"https://www.inoh.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPAC Periodic Table of the Elements and Isotopes (IPTEI)\",\n",
      "        \"url\": \"https://iupac.org/iptei/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPHAR/BPS Guide to PHARMACOLOGY\",\n",
      "        \"url\": \"https://www.guidetopharmacology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"KEGG\",\n",
      "        \"url\": \"https://www.genome.jp/kegg/kegg2.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Kruve Lab, Ionization & Mass Spectrometry, Stockholm University\",\n",
      "        \"url\": \"https://kruvelab.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Lab and Research Safety, University of Minnesota\",\n",
      "        \"url\": \"https://dehs.umn.edu/department-environmental-health-safety/lab-research-safety\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LIPID MAPS\",\n",
      "        \"url\": \"https://lipidmaps.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LOTUS - the natural products occurrence database\",\n",
      "        \"url\": \"https://lotus.naturalproducts.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"MarkerDB\",\n",
      "        \"url\": \"https://markerdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Medical Subject Headings (MeSH)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/mesh\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging and Contrast Agent Database (MICAD)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK5330/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging Database (MOLI)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK5330/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NCI Thesaurus (NCIt)\",\n",
      "        \"url\": \"https://ncit.nci.nih.gov/ncitbrowser/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NEQUIM - Chemoinformatics Group\",\n",
      "        \"url\": \"http://nequim.qui.ufmg.br/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NextMove Software\",\n",
      "        \"url\": \"https://www.nextmovesoftware.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIAID ChemDB\",\n",
      "        \"url\": \"https://chemdb.niaid.nih.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIST Chemistry WebBook\",\n",
      "        \"url\": \"https://webbook.nist.gov/chemistry/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NITE-CMC\",\n",
      "        \"url\": \"https://www.nite.go.jp/chem/english/ghs/ghs_index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Online Mendelian Inheritance in Man (OMIM)\",\n",
      "        \"url\": \"https://omim.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PANTHER\",\n",
      "        \"url\": \"http://www.pantherdb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PathBank\",\n",
      "        \"url\": \"https://pathbank.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pathway Interaction Database\",\n",
      "        \"url\": \"https://wiki.nci.nih.gov/pages/viewpage.action?pageId=315491760\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pfam\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/interpro/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PharmGKB\",\n",
      "        \"url\": \"https://www.pharmgkb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pistoia Alliance DataFAIRy Bioassay Pilot\",\n",
      "        \"url\": \"https://www.pistoiaalliance.org/projects/current-projects/datafairy-project/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Plant Reactome\",\n",
      "        \"url\": \"https://plantreactome.gramene.org/index.php?lang=en\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PlantCyc\",\n",
      "        \"url\": \"https://www.plantcyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Protein Ontology\",\n",
      "        \"url\": \"https://proconsortium.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"RCSB Protein Data Bank (RCSB PDB)\",\n",
      "        \"url\": \"https://www.rcsb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Reactome\",\n",
      "        \"url\": \"https://reactome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Rhea - Annotated Reactions Database\",\n",
      "        \"url\": \"https://www.rhea-db.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"SureChEMBL\",\n",
      "        \"url\": \"https://www.surechembl.org/search/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics Cellosaurus\",\n",
      "        \"url\": \"https://www.cellosaurus.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics ENZYME\",\n",
      "        \"url\": \"https://enzyme.expasy.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Cambridge Structural Database\",\n",
      "        \"url\": \"https://www.ccdc.cam.ac.uk/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The National Institute for Occupational Safety and Health (NIOSH)\",\n",
      "        \"url\": \"https://www.cdc.gov/niosh/npg/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Natural Products Atlas\",\n",
      "        \"url\": \"https://www.npatlas.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The University of Alabama Libraries\",\n",
      "        \"url\": \"https://ir.ua.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Therapeutic Target Database (TTD)\",\n",
      "        \"url\": \"https://db.idrblab.net/ttd/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniCarbKB\",\n",
      "        \"url\": \"http://www.unicarbkb.org\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniProt\",\n",
      "        \"url\": \"https://www.uniprot.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikidata\",\n",
      "        \"url\": \"https://www.wikidata.org/wiki/Wikidata:Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikipedia\",\n",
      "        \"url\": \"https://en.wikipedia.org/wiki/Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"World Register of Marine Species (WoRMS)\",\n",
      "        \"url\": \"https://www.marinespecies.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Yeast Metabolome Database (YMDB)\",\n",
      "        \"url\": \"http://www.ymdb.ca/\"\n",
      "    }\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import json\n",
    "# This URL is the metadata about the data sources in PubChem\n",
    "url = 'https://pubchem.ncbi.nlm.nih.gov/rest/pug/sourcetable/all/JSON/?response_type=display'\n",
    "response = requests.get(url)\n",
    "srcs = response.json()\n",
    "results = []\n",
    "search = 'Curation Efforts'  # i.e., a repository, or other type of data source (this is in index 8 of the data list for each source)\n",
    "rows = srcs['Table']['Row']\n",
    "for row in rows:\n",
    "    if row['Cell'][8].find(search) != -1:\n",
    "        hit = {}\n",
    "        hit.update({'name': row['Cell'][0]})\n",
    "        hit.update({'url': row['Cell'][9]})\n",
    "        results.append(hit)\n",
    "print(json.dumps(results, indent=4))"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Searching FAIR Sharing for datasets\n",
    "FAIRSharing is a database of FAIR resources and per se a database of datasets, however you might find a repository here that\n",
    "has the kind of data you are looking for. The code below a"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "from hiddensettings import *\n",
    "import requests\n",
    "import json\n",
    "# see https://fairsharing.org/API_doc for instructions on how to search the API\n",
    "# user login\n",
    "url = 'https://api.fairsharing.org/users/sign_in'\n",
    "loghdrs = {'Accept': 'application/json','Content-Type': 'application/json'}\n",
    "login = {'user': {'login': fs_user, 'password': fs_pass}}\n",
    "response = requests.request(\"POST\", url, headers=loghdrs, data=json.dumps(login))\n",
    "data = response.json()\n",
    "jwt = data['jwt']\n",
    "srchdrs = {'Accept': 'application/json', 'Content-Type': 'application/json', 'Authorization': \"Bearer {0}\".format(jwt)}\n",
    "searchterm = 'chemistry'\n",
    "searchurl ='https://api.fairsharing.org//search/fairsharing_records?q=' + searchterm\n",
    "search = requests.request(\"POST\", searchurl, headers=srchdrs)\n",
    "hits = json.loads(search.content)\n",
    "for hit in hits['data']:\n",
    "    print(hit['attributes']['name'] + \": \" + hit['attributes']['url'])"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Author: Stuart Chalk\n",
    "Date: Dec. 12, 2022"
   ],
   "metadata": {
    "collapsed": false
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
