

<!DOCTYPE html>


<html >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Finding a Findable Dataset &#8212; IUPAC FAIR Chemistry Cookbook</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/js/matomo.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'manipulations/dataset_search';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Importing Chemical Data into a Spreadsheet" href="excel_import.html" />
    <link rel="prev" title="Computing InChIs" href="computing_inchis.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="None"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
    
    
      
    
    
    <img src="../_static/iupac_and_wf.jpg" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="../_static/iupac_and_wf.jpg" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../about_cookbook.html">About this Cookbook</a></li>
<li class="toctree-l1"><a class="reference internal" href="../use_cookbook.html">How to use the Cookbook</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributions.html">Contribute to the Cookbook!</a></li>
<li class="toctree-l1"><a class="reference internal" href="../feedback.html">Feedback on the Cookbook</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">The FAIR Chemistry Cookbook</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../cooking.html">The Joy of FAIR</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../techniques.html">FAIR Techniques</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../tools.html">Tools &amp; Web Services</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../tools/opsin_api.html">OPSIN: Open Parser for Systematic IUPAC Nomenclature</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tools/opsin_api_jn.html">Using the OPSIN API in Python</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../data_sources.html">Data Sources</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../datasources/goldbook_api.html">Terminology: The IUPAC Gold Book</a></li>
<li class="toctree-l2"><a class="reference internal" href="../datasources/cas_commonchem.html">CAS Common Chemistry API</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../datasources/pubchem.html">Data Repository: PubChem</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3 has-children"><a class="reference internal" href="../datasources/pubchem_pugrest.html">PubChem PUG-REST</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="../datasources/pubchem_pugrest1.html">Accessing PubChem through PUG-REST - Part I</a></li>
<li class="toctree-l4"><a class="reference internal" href="../datasources/pubchem_pugrest2.html">Accessing PubChem through PUG-REST: Part II</a></li>
<li class="toctree-l4"><a class="reference internal" href="../datasources/pubchem_pugrest3.html">Accessing PubChem through PUG-REST: Part III</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../datasources/pubchem_pugview.html">PubChem PUG-View</a></li>
<li class="toctree-l3"><a class="reference internal" href="../datasources/pubchem_ptable.html">PubChem Periodic Table</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../manipulations.html">Data Manipulations</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="computing_inchis.html">Computing InChIs</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Finding a Findable Dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="excel_import.html">Importing Chemical Data into a Spreadsheet</a></li>
<li class="toctree-l2"><a class="reference internal" href="gsheets_citations.html">Importing JSON Citation Data from an API into Google Sheets</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../usecases.html">Use Cases &amp; Workflows</a><input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../usecases/iupac_pkadata.html">Reuse of Curated Chemical Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../usecases/uvvis_metadata.html">Structured UV/Vis (meta)data for small molecule characterisation</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Reference</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../about_project.html">About this project</a><input class="toctree-checkbox" id="toctree-checkbox-8" name="toctree-checkbox-8" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-8"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../contributors.html">Project Contributors</a></li>
<li class="toctree-l2"><a class="reference internal" href="../development.html">Cookbook development</a></li>
<li class="toctree-l2"><a class="reference internal" href="../worldfair.html">WordFAIR Chemistry</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../glossary.html">Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../bibliography.html">Bibliography</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-launch-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Launch interactive content">
    <i class="fas fa-rocket"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://mybinder.org/v2/gh/IUPAC/WFChemCookbook/main?urlpath=tree/book/manipulations/dataset_search.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch onBinder"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img src="../_static/images/logo_binder.svg">
  </span>
<span class="btn__text-container">Binder</span>
</a>
</li>
      
      
      
      
      <li><a href="https://colab.research.google.com/github/IUPAC/WFChemCookbook/blob/main/book/manipulations/dataset_search.ipynb" target="_blank"
   class="btn btn-sm dropdown-item"
   title="Launch onColab"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  
    <img src="../_static/images/logo_colab.png">
  </span>
<span class="btn__text-container">Colab</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/IUPAC/WFChemCookbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/IUPAC/WFChemCookbook/issues/new?title=Issue%20on%20page%20%2Fmanipulations/dataset_search.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/manipulations/dataset_search.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Finding a Findable Dataset</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#scenario">Scenario</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-searching-pubchem-for-datasets">Step 1 - Searching PubChem for datasets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-the-python-functions">1.1 - Load the Python functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#search-for-sources-that-have-curation-efforts">1.2 - Search for sources that have ‘curation efforts’</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-searching-fair-sharing-for-datasets">Step 2 - Searching FAIR Sharing for datasets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#authentication-to-the-fairsharing-api">2.1 - Authentication to the FAIRSharing API</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#make-the-api-request">2.2 - Make the API request</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#output-the-data-in-a-presentable-format">2.3 - Output the data in a presentable format</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section id="finding-a-findable-dataset">
<h1>Finding a Findable Dataset<a class="headerlink" href="#finding-a-findable-dataset" title="Permalink to this headline">#</a></h1>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header">
About this interactive <img alt="icons" src="../_images/rocket1.png" /> recipe<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<ul class="simple">
<li><p class="sd-card-text">Author(s): <a class="reference external" href="https://orcid.org/0000-0002-0703-7776">Stuart Chalk</a></p></li>
<li><p class="sd-card-text">Topic(s): How and where to find a ‘findable’ chemical dataset</p></li>
<li><p class="sd-card-text">Format(s): Interactive Jupyter Notebook (Python)</p></li>
<li><p class="sd-card-text">Scenario(s): You are looking for research data to complement your compare with your own data</p></li>
<li><p class="sd-card-text">Skill(s): You should be familiar with</p>
<ul>
<li><p class="sd-card-text"><a class="reference external" href="https://www.ibm.com/topics/api">Application Programming Interfaces (APIs)</a></p></li>
<li><p class="sd-card-text"><a class="reference external" href="https://doi.org/10.1515/pac-2021-3013">Working with a data model</a></p></li>
<li><p class="sd-card-text"><a class="reference external" href="https://www.youtube.com/watch?v=iiADhChRriM">Introductory JSON</a></p></li>
</ul>
</li>
<li><p class="sd-card-text">Learning outcomes: After completing this example you should understand:</p>
<ul>
<li><p class="sd-card-text">How to make a request to a website using the Python ‘requests’ functionality</p></li>
<li><p class="sd-card-text">Retrieve data in JSON format and how to parse it (knowing the data model)</p></li>
<li><p class="sd-card-text">How to store confidential data in a remote file</p></li>
<li><p class="sd-card-text">How programmtically you can authenticate to an API (one of many ways)</p></li>
</ul>
</li>
<li><p class="sd-card-text">Citation: ‘Finding a Findable Dataset’, Stuart Chalk, The IUPAC FAIR Chemistry Cookbook, Contributed: 2024-02-14 <a class="reference external" href="https://w3id.org/ifcc/IFCC013">https://w3id.org/ifcc/IFCC013</a>.</p></li>
<li><p class="sd-card-text">Reuse: This notebook is made available under a <a class="reference external" href="https://creativecommons.org/licenses/by/4.0/">CC-BY-4.0</a> license.</p></li>
</ul>
</div>
</details><section id="scenario">
<h2>Scenario<a class="headerlink" href="#scenario" title="Permalink to this headline">#</a></h2>
<p>Our group has a set of thermophysical data on over 8000 chemical substances.  We want to integrate into this dataset another physical property dataset so that we can do an analysis of the correlations of the thermophysical data with the chosen physical property of the substances (that are common to both sets).</p>
<p>Criteria for picking the physical property dataset: high quality, trusted, large, available with an open license, so I can publish the results and make the derived dataset open.</p>
<ul class="simple">
<li><p><strong>High quality means</strong>: unambiguous identification of each chemical substance, enough contextual information (metadata) to make the values scientifically useful, i.e., at least the composition of the solvent, the temperature and for volatile substances the pressure.</p></li>
<li><p><strong>Trusted means</strong>: the provenance chain is reported with the data, and it shows that the data comes from a reputable source(s) and any aggregation and/or processing is documented in enough detail that the community can understand how the dataset has been created/provided.</p></li>
</ul>
</section>
<section id="step-1-searching-pubchem-for-datasets">
<h2>Step 1 - Searching PubChem for datasets<a class="headerlink" href="#step-1-searching-pubchem-for-datasets" title="Permalink to this headline">#</a></h2>
<p>Pubchem houses a lot of data about chemical substances, compounds and bioassays.  Over time external organizations have worked with PubChem to include data, in one of a couple of ways:</p>
<ul class="simple">
<li><p>data that has been integrated into PubChem pages (e.g., <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/source/941">CCDC</a> -&gt; <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/compound/241">example</a>)</p></li>
<li><p>data that is not available in a PubChem page but is available via the data sources section of the site as ‘annotations’ (e.g. <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/source/15751">RCSB PDB</a> -&gt; <a class="reference external" href="https://pubchem.ncbi.nlm.nih.gov/source/15751#data=Annotations">Example</a>)</p></li>
</ul>
<p>The data available is may not be structured and or clearly described, however if the source has a website with an API then you are likely to get better quality metadata from the linked site.</p>
<section id="load-the-python-functions">
<h3>1.1 - Load the Python functions<a class="headerlink" href="#load-the-python-functions" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># as these are direct imports (they do not reference a Python package) they are built into Python</span>
<span class="kn">import</span> <span class="nn">requests</span>
<span class="kn">import</span> <span class="nn">json</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="search-for-sources-that-have-curation-efforts">
<h3>1.2 - Search for sources that have ‘curation efforts’<a class="headerlink" href="#search-for-sources-that-have-curation-efforts" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># This URL is the metadata about the data sources in PubChem</span>
<span class="n">url</span> <span class="o">=</span> <span class="s1">&#39;https://pubchem.ncbi.nlm.nih.gov/rest/pug/sourcetable/all/JSON/?response_type=display&#39;</span>
<span class="n">response</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">url</span><span class="p">)</span>
<span class="n">srcs</span> <span class="o">=</span> <span class="n">response</span><span class="o">.</span><span class="n">json</span><span class="p">()</span>
<span class="n">results</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">search</span> <span class="o">=</span> <span class="s1">&#39;Curation Efforts&#39;</span>  <span class="c1"># i.e., a repository, or other type of data source (this is in index 8 of the data list for each source)</span>
<span class="n">rows</span> <span class="o">=</span> <span class="n">srcs</span><span class="p">[</span><span class="s1">&#39;Table&#39;</span><span class="p">][</span><span class="s1">&#39;Row&#39;</span><span class="p">]</span>
<span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">rows</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;Cell&#39;</span><span class="p">][</span><span class="mi">8</span><span class="p">]</span><span class="o">.</span><span class="n">find</span><span class="p">(</span><span class="n">search</span><span class="p">)</span> <span class="o">!=</span> <span class="o">-</span><span class="mi">1</span><span class="p">:</span>
        <span class="n">hit</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="n">hit</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;Cell&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">]})</span>
        <span class="n">hit</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s1">&#39;url&#39;</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;Cell&#39;</span><span class="p">][</span><span class="mi">9</span><span class="p">]})</span>
        <span class="n">results</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">hit</span><span class="p">)</span>
<span class="c1"># when printed this is a scrollable list of many entries</span>
<span class="nb">print</span><span class="p">(</span><span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span><span class="n">results</span><span class="p">,</span> <span class="n">indent</span><span class="o">=</span><span class="mi">4</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[
    {
        &quot;name&quot;: &quot;Agency for Toxic Substances and Disease Registry (ATSDR)&quot;,
        &quot;url&quot;: &quot;https://www.atsdr.cdc.gov/&quot;
    },
    {
        &quot;name&quot;: &quot;Alliance of Genome Resources&quot;,
        &quot;url&quot;: &quot;https://www.alliancegenome.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Athena Minerals&quot;,
        &quot;url&quot;: &quot;https://athena.unige.ch/athena/mineral/mineral.html&quot;
    },
    {
        &quot;name&quot;: &quot;Barrie Walker, BARK Information Services&quot;,
        &quot;url&quot;: &quot;https://uk.linkedin.com/in/barrie-walker-85b4a510&quot;
    },
    {
        &quot;name&quot;: &quot;BindingDB&quot;,
        &quot;url&quot;: &quot;https://www.bindingdb.org/rwd/bind/&quot;
    },
    {
        &quot;name&quot;: &quot;BioCyc&quot;,
        &quot;url&quot;: &quot;https://biocyc.org/&quot;
    },
    {
        &quot;name&quot;: &quot;BioGRID&quot;,
        &quot;url&quot;: &quot;https://thebiogrid.org/&quot;
    },
    {
        &quot;name&quot;: &quot;BRENDA: Enzyme Functional Data&quot;,
        &quot;url&quot;: &quot;https://www.brenda-enzymes.org/&quot;
    },
    {
        &quot;name&quot;: &quot;CAMEO Chemicals&quot;,
        &quot;url&quot;: &quot;https://cameochemicals.noaa.gov/&quot;
    },
    {
        &quot;name&quot;: &quot;Catalogue of Life (COL)&quot;,
        &quot;url&quot;: &quot;https://www.catalogueoflife.org/&quot;
    },
    {
        &quot;name&quot;: &quot;CCSbase&quot;,
        &quot;url&quot;: &quot;https://ccsbase.net/&quot;
    },
    {
        &quot;name&quot;: &quot;Cell Line Ontology (CLO)&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;Cell Ontology (CL)&quot;,
        &quot;url&quot;: &quot;https://obophenotype.github.io/cell-ontology/&quot;
    },
    {
        &quot;name&quot;: &quot;ChEBI&quot;,
        &quot;url&quot;: &quot;https://www.ebi.ac.uk/chebi/&quot;
    },
    {
        &quot;name&quot;: &quot;ChEMBL&quot;,
        &quot;url&quot;: &quot;https://www.ebi.ac.uk/chembl/&quot;
    },
    {
        &quot;name&quot;: &quot;Chemical Probes Portal&quot;,
        &quot;url&quot;: &quot;https://www.chemicalprobes.org/&quot;
    },
    {
        &quot;name&quot;: &quot;ChemIDplus&quot;,
        &quot;url&quot;: &quot;https://pubchem.ncbi.nlm.nih.gov/source/ChemIDplus&quot;
    },
    {
        &quot;name&quot;: &quot;Chemoproteomic Metabolic Pathway Resource, Scripps University&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;Chris Southan&quot;,
        &quot;url&quot;: &quot;https://www.ed.ac.uk/medicine-vet-medicine&quot;
    },
    {
        &quot;name&quot;: &quot;Comparative Toxicogenomics Database (CTD)&quot;,
        &quot;url&quot;: &quot;http://ctdbase.org/&quot;
    },
    {
        &quot;name&quot;: &quot;COVID-19 Disease Map&quot;,
        &quot;url&quot;: &quot;https://covid.pages.uni.lu/&quot;
    },
    {
        &quot;name&quot;: &quot;Database of Interacting Proteins (DIP)&quot;,
        &quot;url&quot;: &quot;https://dip.doe-mbi.ucla.edu/&quot;
    },
    {
        &quot;name&quot;: &quot;Drug Gene Interaction database (DGIdb)&quot;,
        &quot;url&quot;: &quot;http://www.dgidb.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Drug Induced Liver Injury Rank (DILIrank) Dataset&quot;,
        &quot;url&quot;: &quot;https://www.fda.gov/science-research/liver-toxicity-knowledge-base-ltkb/drug-induced-liver-injury-rank-dilirank-dataset&quot;
    },
    {
        &quot;name&quot;: &quot;DrugBank&quot;,
        &quot;url&quot;: &quot;https://www.drugbank.ca/&quot;
    },
    {
        &quot;name&quot;: &quot;DrugCentral&quot;,
        &quot;url&quot;: &quot;http://drugcentral.org&quot;
    },
    {
        &quot;name&quot;: &quot;E. coli Metabolome Database (ECMDB)&quot;,
        &quot;url&quot;: &quot;https://ecmdb.ca/&quot;
    },
    {
        &quot;name&quot;: &quot;Egon Willighagen, Department of Bioinformatics - BiGCaT, Maastricht University&quot;,
        &quot;url&quot;: &quot;https://www.maastrichtuniversity.nl/el-willighagen&quot;
    },
    {
        &quot;name&quot;: &quot;EPA Air Toxics&quot;,
        &quot;url&quot;: &quot;https://www3.epa.gov/ttn/atw/index.html&quot;
    },
    {
        &quot;name&quot;: &quot;EPA DSSTox&quot;,
        &quot;url&quot;: &quot;https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database&quot;
    },
    {
        &quot;name&quot;: &quot;FDA Global Substance Registration System (GSRS)&quot;,
        &quot;url&quot;: &quot;https://www.fda.gov/industry/fda-data-standards-advisory-board/fdas-global-substance-registration-system&quot;
    },
    {
        &quot;name&quot;: &quot;FDA Orange Book&quot;,
        &quot;url&quot;: &quot;https://www.fda.gov/drugs/drug-approvals-and-databases/approved-drug-products-therapeutic-equivalence-evaluations-orange-book&quot;
    },
    {
        &quot;name&quot;: &quot;FDA Pharm Classes&quot;,
        &quot;url&quot;: &quot;https://www.fda.gov/industry/structured-product-labeling-resources/pharmacologic-class&quot;
    },
    {
        &quot;name&quot;: &quot;FlyBase&quot;,
        &quot;url&quot;: &quot;https://flybase.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Gene Curation Coalition (GenCC)&quot;,
        &quot;url&quot;: &quot;https://thegencc.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Gene Ontology (GO)&quot;,
        &quot;url&quot;: &quot;http://geneontology.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Gloriam Group: Computational Receptor Biology - Department of Drug Design and Pharmacology, University of Copenhagen&quot;,
        &quot;url&quot;: &quot;https://drug.ku.dk/research-sections/pharmaceutical-informatics/gloriam-group/&quot;
    },
    {
        &quot;name&quot;: &quot;GlycoNAVI&quot;,
        &quot;url&quot;: &quot;https://glyconavi.org/&quot;
    },
    {
        &quot;name&quot;: &quot;GlyGen&quot;,
        &quot;url&quot;: &quot;https://www.glygen.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Haz-Map, Information on Hazardous Chemicals and Occupational Diseases&quot;,
        &quot;url&quot;: &quot;https://haz-map.com/&quot;
    },
    {
        &quot;name&quot;: &quot;Hazardous Chemical Information System (HCIS), Safe Work Australia&quot;,
        &quot;url&quot;: &quot;http://hcis.safeworkaustralia.gov.au/&quot;
    },
    {
        &quot;name&quot;: &quot;Hazardous Substances Data Bank (HSDB)&quot;,
        &quot;url&quot;: &quot;https://www.nlm.nih.gov/toxnet/index.html&quot;
    },
    {
        &quot;name&quot;: &quot;HUGO Gene Nomenclature Committee (HGNC)&quot;,
        &quot;url&quot;: &quot;https://www.genenames.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Human Metabolome Database (HMDB)&quot;,
        &quot;url&quot;: &quot;https://hmdb.ca/&quot;
    },
    {
        &quot;name&quot;: &quot;Human Protein Atlas (HPA)&quot;,
        &quot;url&quot;: &quot;https://www.proteinatlas.org/&quot;
    },
    {
        &quot;name&quot;: &quot;ILO-WHO International Chemical Safety Cards (ICSCs)&quot;,
        &quot;url&quot;: &quot;https://www.ilo.org/safework/info/publications/WCMS_113134/lang--en/index.htm&quot;
    },
    {
        &quot;name&quot;: &quot;INOH&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;IntAct Molecular Interaction Database&quot;,
        &quot;url&quot;: &quot;https://www.ebi.ac.uk/intact/&quot;
    },
    {
        &quot;name&quot;: &quot;InterPro&quot;,
        &quot;url&quot;: &quot;https://www.ebi.ac.uk/interpro/&quot;
    },
    {
        &quot;name&quot;: &quot;IUPAC Periodic Table of the Elements and Isotopes (IPTEI)&quot;,
        &quot;url&quot;: &quot;https://iupac.org/iptei/&quot;
    },
    {
        &quot;name&quot;: &quot;IUPHAR/BPS Guide to PHARMACOLOGY&quot;,
        &quot;url&quot;: &quot;https://www.guidetopharmacology.org/&quot;
    },
    {
        &quot;name&quot;: &quot;KEGG&quot;,
        &quot;url&quot;: &quot;https://www.genome.jp/kegg/kegg2.html&quot;
    },
    {
        &quot;name&quot;: &quot;KNApSAcK Species-Metabolite Database&quot;,
        &quot;url&quot;: &quot;http://www.knapsackfamily.com/KNApSAcK/&quot;
    },
    {
        &quot;name&quot;: &quot;Kruve Lab, Ionization &amp; Mass Spectrometry, Stockholm University&quot;,
        &quot;url&quot;: &quot;https://kruvelab.com/&quot;
    },
    {
        &quot;name&quot;: &quot;Lab and Research Safety, University of Minnesota&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;LIPID MAPS&quot;,
        &quot;url&quot;: &quot;https://lipidmaps.org/&quot;
    },
    {
        &quot;name&quot;: &quot;LOTUS - the natural products occurrence database&quot;,
        &quot;url&quot;: &quot;https://lotus.naturalproducts.net/&quot;
    },
    {
        &quot;name&quot;: &quot;MarkerDB&quot;,
        &quot;url&quot;: &quot;https://markerdb.ca/&quot;
    },
    {
        &quot;name&quot;: &quot;Medical Subject Headings (MeSH)&quot;,
        &quot;url&quot;: &quot;https://www.ncbi.nlm.nih.gov/mesh&quot;
    },
    {
        &quot;name&quot;: &quot;Molecular Imaging and Contrast Agent Database (MICAD)&quot;,
        &quot;url&quot;: &quot;https://www.ncbi.nlm.nih.gov/books/NBK5330/&quot;
    },
    {
        &quot;name&quot;: &quot;Molecular Imaging Database (MOLI)&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;Mouse Genome Informatics (MGI)&quot;,
        &quot;url&quot;: &quot;https://www.informatics.jax.org/&quot;
    },
    {
        &quot;name&quot;: &quot;NCI Thesaurus (NCIt)&quot;,
        &quot;url&quot;: &quot;https://ncit.nci.nih.gov/ncitbrowser/&quot;
    },
    {
        &quot;name&quot;: &quot;NEQUIM - Chemoinformatics Group&quot;,
        &quot;url&quot;: &quot;http://nequim.qui.ufmg.br/&quot;
    },
    {
        &quot;name&quot;: &quot;NextMove Software&quot;,
        &quot;url&quot;: &quot;https://www.nextmovesoftware.com/&quot;
    },
    {
        &quot;name&quot;: &quot;NIAID ChemDB&quot;,
        &quot;url&quot;: &quot;https://chemdb.niaid.nih.gov/&quot;
    },
    {
        &quot;name&quot;: &quot;NIST Chemistry WebBook&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;NITE-CMC&quot;,
        &quot;url&quot;: &quot;https://www.nite.go.jp/chem/english/ghs/ghs_index.html&quot;
    },
    {
        &quot;name&quot;: &quot;Online Mendelian Inheritance in Man (OMIM)&quot;,
        &quot;url&quot;: &quot;https://omim.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Open Targets&quot;,
        &quot;url&quot;: &quot;https://www.opentargets.org/&quot;
    },
    {
        &quot;name&quot;: &quot;PANTHER&quot;,
        &quot;url&quot;: &quot;http://www.pantherdb.org/&quot;
    },
    {
        &quot;name&quot;: &quot;PathBank&quot;,
        &quot;url&quot;: &quot;https://pathbank.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Pathway Interaction Database&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;Pfam&quot;,
        &quot;url&quot;: &quot;https://www.ebi.ac.uk/interpro/&quot;
    },
    {
        &quot;name&quot;: &quot;PharmGKB&quot;,
        &quot;url&quot;: &quot;https://www.pharmgkb.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Pharos&quot;,
        &quot;url&quot;: &quot;https://pharos.nih.gov/&quot;
    },
    {
        &quot;name&quot;: &quot;Pistoia Alliance DataFAIRy Bioassay Pilot&quot;,
        &quot;url&quot;: &quot;https://www.pistoiaalliance.org/projects/current-projects/datafairy-project/&quot;
    },
    {
        &quot;name&quot;: &quot;Plant Reactome&quot;,
        &quot;url&quot;: &quot;https://plantreactome.gramene.org/index.php?lang=en&quot;
    },
    {
        &quot;name&quot;: &quot;PlantCyc&quot;,
        &quot;url&quot;: &quot;https://www.plantcyc.org/&quot;
    },
    {
        &quot;name&quot;: &quot;PomBase: Fission Yeast Resource&quot;,
        &quot;url&quot;: &quot;https://www.pombase.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Protein Ontology&quot;,
        &quot;url&quot;: &quot;https://proconsortium.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Rat Genome Database (RGD)&quot;,
        &quot;url&quot;: &quot;https://rgd.mcw.edu/&quot;
    },
    {
        &quot;name&quot;: &quot;RCSB Protein Data Bank (RCSB PDB)&quot;,
        &quot;url&quot;: &quot;https://www.rcsb.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Reactome&quot;,
        &quot;url&quot;: &quot;https://reactome.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Rhea - Annotated Reactions Database&quot;,
        &quot;url&quot;: &quot;https://www.rhea-db.org/&quot;
    },
    {
        &quot;name&quot;: &quot;RRUFF Project&quot;,
        &quot;url&quot;: &quot;https://rruff.info/&quot;
    },
    {
        &quot;name&quot;: &quot;Saccharomyces Genome Database (SGD)&quot;,
        &quot;url&quot;: &quot;https://www.yeastgenome.org/&quot;
    },
    {
        &quot;name&quot;: &quot;SureChEMBL&quot;,
        &quot;url&quot;: &quot;https://www.surechembl.org/search/&quot;
    },
    {
        &quot;name&quot;: &quot;Swiss Institute of Bioinformatics Bgee&quot;,
        &quot;url&quot;: &quot;https://www.bgee.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Swiss Institute of Bioinformatics Cellosaurus&quot;,
        &quot;url&quot;: &quot;https://www.cellosaurus.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Swiss Institute of Bioinformatics ENZYME&quot;,
        &quot;url&quot;: &quot;https://enzyme.expasy.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Swiss Institute of Bioinformatics neXtProt&quot;,
        &quot;url&quot;: &quot;https://www.nextprot.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Symbol Nomenclature for Glycans (SNFG) Reference Collection&quot;,
        &quot;url&quot;: &quot;https://www.ncbi.nlm.nih.gov/glycans/snfg.html&quot;
    },
    {
        &quot;name&quot;: &quot;The Cambridge Structural Database&quot;,
        &quot;url&quot;: &quot;https://www.ccdc.cam.ac.uk/&quot;
    },
    {
        &quot;name&quot;: &quot;The National Institute for Occupational Safety and Health (NIOSH)&quot;,
        &quot;url&quot;: &quot;https://www.cdc.gov/niosh/npg/&quot;
    },
    {
        &quot;name&quot;: &quot;The Natural Products Atlas&quot;,
        &quot;url&quot;: &quot;https://www.npatlas.org/&quot;
    },
    {
        &quot;name&quot;: &quot;The University of Alabama Libraries&quot;,
        &quot;url&quot;: &quot;https://ir.ua.edu/&quot;
    },
    {
        &quot;name&quot;: &quot;The Zebrafish Information Network (ZFIN)&quot;,
        &quot;url&quot;: &quot;https://zfin.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Therapeutic Target Database (TTD)&quot;,
        &quot;url&quot;: &quot;https://db.idrblab.net/ttd/&quot;
    },
    {
        &quot;name&quot;: &quot;Toxin and Toxin Target Database (T3DB)&quot;,
        &quot;url&quot;: &quot;http://www.t3db.ca/&quot;
    },
    {
        &quot;name&quot;: &quot;UniCarbKB&quot;,
        &quot;url&quot;: &quot;&quot;
    },
    {
        &quot;name&quot;: &quot;UniProt&quot;,
        &quot;url&quot;: &quot;https://www.uniprot.org/&quot;
    },
    {
        &quot;name&quot;: &quot;UVCBs and Multi-Constituent Substances (MCS) Committee&quot;,
        &quot;url&quot;: &quot;https://hesiglobal.org/uvcb/&quot;
    },
    {
        &quot;name&quot;: &quot;VEuPathDB: The Eukaryotic Pathogen, Vector and Host Informatics Resource&quot;,
        &quot;url&quot;: &quot;https://veupathdb.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Wikidata&quot;,
        &quot;url&quot;: &quot;https://www.wikidata.org/wiki/Wikidata:Main_Page&quot;
    },
    {
        &quot;name&quot;: &quot;Wikipedia&quot;,
        &quot;url&quot;: &quot;https://en.wikipedia.org/wiki/Main_Page&quot;
    },
    {
        &quot;name&quot;: &quot;World Register of Marine Species (WoRMS)&quot;,
        &quot;url&quot;: &quot;https://www.marinespecies.org/&quot;
    },
    {
        &quot;name&quot;: &quot;WormBase&quot;,
        &quot;url&quot;: &quot;https://wormbase.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Xenbase&quot;,
        &quot;url&quot;: &quot;https://www.xenbase.org/&quot;
    },
    {
        &quot;name&quot;: &quot;Yeast Metabolome Database (YMDB)&quot;,
        &quot;url&quot;: &quot;http://www.ymdb.ca/&quot;
    }
]
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="step-2-searching-fair-sharing-for-datasets">
<h2>Step 2 - Searching FAIR Sharing for datasets<a class="headerlink" href="#step-2-searching-fair-sharing-for-datasets" title="Permalink to this headline">#</a></h2>
<p>FAIRSharing is a database of FAIR resources and per se a database of datasets, however you might find a repository here that
has the kind of data you are looking for. The code below accesses the FAIR sharing API so search for ‘chemistry’ (or other term) related resources.</p>
<p><em>Note: To use the code below please go to <a class="reference external" href="https://fairsharing.org/accounts/signup?ref=iupacfaircookbook">https://fairsharing.org/accounts/signup</a>, create an account and then enter your username and password in the quotes for ‘fs_user’ and ‘fs_pass’ below.</em></p>
<section id="authentication-to-the-fairsharing-api">
<h3>2.1 - Authentication to the FAIRSharing API<a class="headerlink" href="#authentication-to-the-fairsharing-api" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># see https://fairsharing.org/API_doc for instructions on how to search the API</span>
<span class="c1"># user login</span>
<span class="n">fs_user</span> <span class="o">=</span> <span class="s2">&quot;ChemCookbook&quot;</span>
<span class="n">fs_pass</span> <span class="o">=</span> <span class="s2">&quot;ydt_wdh_MRD*qut5xvq&quot;</span>
<span class="n">url</span> <span class="o">=</span> <span class="s1">&#39;https://api.fairsharing.org/users/sign_in&#39;</span>
<span class="n">loghdrs</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Accept&#39;</span><span class="p">:</span> <span class="s1">&#39;application/json&#39;</span><span class="p">,</span><span class="s1">&#39;Content-Type&#39;</span><span class="p">:</span> <span class="s1">&#39;application/json&#39;</span><span class="p">}</span>
<span class="n">login</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;user&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;login&#39;</span><span class="p">:</span> <span class="n">fs_user</span><span class="p">,</span> <span class="s1">&#39;password&#39;</span><span class="p">:</span> <span class="n">fs_pass</span><span class="p">}}</span>
<span class="n">response</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">request</span><span class="p">(</span><span class="s2">&quot;POST&quot;</span><span class="p">,</span> <span class="n">url</span><span class="p">,</span> <span class="n">headers</span><span class="o">=</span><span class="n">loghdrs</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span><span class="n">login</span><span class="p">))</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">response</span><span class="o">.</span><span class="n">json</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>{&#39;success&#39;: True, &#39;jwt&#39;: &#39;eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI0YjVkNzQyMi1hOTg3LTRlZWYtYjQyYi1hN2U3Yjc2MTM0ZTEiLCJzdWIiOiI4NTA5Iiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNzA4MDkxOTA0LCJleHAiOjE3MDgxNzgzMDR9.HzRbgJOlDiC4vvgbsPV10_ub4Bx4rBgsUDwsIarI4pY&#39;, &#39;username&#39;: &#39;ChemCookbook&#39;, &#39;id&#39;: 8509, &#39;role&#39;: &#39;user&#39;, &#39;profile_type&#39;: &#39;none&#39;, &#39;watched_records&#39;: [], &#39;is_curator&#39;: False, &#39;is_super_curator&#39;: False, &#39;third_party&#39;: False, &#39;expiry&#39;: 1708178304, &#39;message&#39;: &#39;Authentication successful&#39;}
</pre></div>
</div>
</div>
</div>
</section>
<section id="make-the-api-request">
<h3>2.2 - Make the API request<a class="headerlink" href="#make-the-api-request" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># in order to authenticate when making an API request the &#39;jwt&#39; code above must</span>
<span class="c1"># be included in the JSON headers (see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields)</span>
<span class="n">jwt</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;jwt&#39;</span><span class="p">]</span>
<span class="n">srchdrs</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Accept&#39;</span><span class="p">:</span> <span class="s1">&#39;application/json&#39;</span><span class="p">,</span> <span class="s1">&#39;Content-Type&#39;</span><span class="p">:</span> <span class="s1">&#39;application/json&#39;</span><span class="p">,</span> <span class="s1">&#39;Authorization&#39;</span><span class="p">:</span> <span class="s2">&quot;Bearer </span><span class="si">{0}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">jwt</span><span class="p">)}</span>
<span class="n">searchterm</span> <span class="o">=</span> <span class="s1">&#39;chemistry&#39;</span>
<span class="n">searchurl</span> <span class="o">=</span><span class="s1">&#39;https://api.fairsharing.org/search/fairsharing_records?q=&#39;</span> <span class="o">+</span> <span class="n">searchterm</span>
<span class="n">search</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">request</span><span class="p">(</span><span class="s2">&quot;POST&quot;</span><span class="p">,</span> <span class="n">searchurl</span><span class="p">,</span> <span class="n">headers</span><span class="o">=</span><span class="n">srchdrs</span><span class="p">)</span>
<span class="n">hits</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">search</span><span class="o">.</span><span class="n">content</span><span class="p">)</span>
<span class="c1"># this prints out the raw JSON for the first entry (the &#39;data&#39; entry is a JSON list)</span>
<span class="c1"># that is returned from the API request (formatted nicely, which means its on many lines)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span><span class="n">hits</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">indent</span><span class="o">=</span><span class="mi">4</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>{
    &quot;id&quot;: &quot;1443&quot;,
    &quot;type&quot;: &quot;fairsharing_records&quot;,
    &quot;attributes&quot;: {
        &quot;created_at&quot;: &quot;2021-06-29T14:47:39.000Z&quot;,
        &quot;updated_at&quot;: &quot;2023-03-15T08:08:11.434Z&quot;,
        &quot;metadata&quot;: {
            &quot;doi&quot;: &quot;10.25504/FAIRsharing.cb1adb&quot;,
            &quot;name&quot;: &quot;Portable reduced-precision binary format for trajectories produced by GROMACS package.&quot;,
            &quot;status&quot;: &quot;ready&quot;,
            &quot;contacts&quot;: [
                {
                    &quot;contact_name&quot;: &quot;Adam Hospital&quot;,
                    &quot;contact_email&quot;: &quot;adam.hospital@irbbarcelona.org&quot;,
                    &quot;contact_orcid&quot;: &quot;0000-0002-8291-8071&quot;
                }
            ],
            &quot;homepage&quot;: &quot;https://manual.gromacs.org/documentation/2021/reference-manual/file-formats.html#xtc&quot;,
            &quot;citations&quot;: [],
            &quot;identifier&quot;: 1443,
            &quot;description&quot;: &quot;The XTC format is a portable binary format for trajectories produced by GROMACS package. It uses the External Data Representation (xdr) routines for writing and reading data which were created for the Unix Network File System (NFS). Trajectories are written in XTC files using a reduced precision (lossy) algorithm which works multiplying the coordinates by a scaling factor (typically 1000), so converting them to pm (GROMACS standard distance unit is nm). This allows an integer rounding of the values. Several other tricks are performed, such as making use of atom proximity information: atoms close in sequence are usually close in space (e.g. water molecules). That makes XTC format the most efficient in terms of disk usage, in most cases reducing by a factor of 2 the size of any other binary trajectory format.&quot;,
            &quot;abbreviation&quot;: &quot;xtc&quot;,
            &quot;support_links&quot;: [
                {
                    &quot;url&quot;: &quot;https://manual.gromacs.org/documentation/2021/reference-manual/file-formats.html#xtc&quot;,
                    &quot;name&quot;: &quot;GROMACS reference manual&quot;,
                    &quot;type&quot;: &quot;Help documentation&quot;
                }
            ],
            &quot;year_creation&quot;: 1995,
            &quot;associated_tools&quot;: [
                {
                    &quot;url&quot;: &quot;https://www.gromacs.org/&quot;,
                    &quot;name&quot;: &quot;GROMACS 2021&quot;
                }
            ],
            &quot;deprecation_reason&quot;: null
        },
        &quot;legacy_ids&quot;: [
            &quot;bsg-001620&quot;,
            &quot;bsg-s001620&quot;
        ],
        &quot;name&quot;: &quot;FAIRsharing record for: Portable reduced-precision binary format for trajectories produced by GROMACS package.&quot;,
        &quot;abbreviation&quot;: &quot;xtc&quot;,
        &quot;url&quot;: &quot;https://fairsharing.org/10.25504/FAIRsharing.cb1adb&quot;,
        &quot;doi&quot;: &quot;10.25504/FAIRsharing.cb1adb&quot;,
        &quot;fairsharing_licence&quot;: &quot;https://creativecommons.org/licenses/by-sa/4.0/. Please link to https://fairsharing.org and https://api.fairsharing.org/img/fairsharing-attribution.svg for attribution.&quot;,
        &quot;description&quot;: &quot;This FAIRsharing record describes: The XTC format is a portable binary format for trajectories produced by GROMACS package. It uses the External Data Representation (xdr) routines for writing and reading data which were created for the Unix Network File System (NFS). Trajectories are written in XTC files using a reduced precision (lossy) algorithm which works multiplying the coordinates by a scaling factor (typically 1000), so converting them to pm (GROMACS standard distance unit is nm). This allows an integer rounding of the values. Several other tricks are performed, such as making use of atom proximity information: atoms close in sequence are usually close in space (e.g. water molecules). That makes XTC format the most efficient in terms of disk usage, in most cases reducing by a factor of 2 the size of any other binary trajectory format.&quot;,
        &quot;linked_records&quot;: [],
        &quot;linking_records&quot;: [],
        &quot;fairsharing_registry&quot;: &quot;Standard&quot;,
        &quot;record_type&quot;: &quot;model_and_format&quot;,
        &quot;subjects&quot;: [
            &quot;Computational Chemistry&quot;
        ],
        &quot;domains&quot;: [],
        &quot;taxonomies&quot;: [
            &quot;Not applicable&quot;
        ],
        &quot;user_defined_tags&quot;: [],
        &quot;countries&quot;: [
            &quot;Netherlands&quot;,
            &quot;Sweden&quot;
        ],
        &quot;publications&quot;: [
            {
                &quot;id&quot;: 2540,
                &quot;pubmed_id&quot;: null,
                &quot;title&quot;: &quot;Methods and Techniques in Computational Chemistry: METECC-95&quot;,
                &quot;year&quot;: 1995,
                &quot;url&quot;: &quot;https://www.semanticscholar.org/paper/Methods-and-techniques-in-computational-chemistry-%3A-Clementi-Corongiu/40385d32372c7c8af5dd4fe1ee442a47bc8ad87d&quot;,
                &quot;authors&quot;: &quot;D. Green, K. Meacham, M. Surridge, F. van Hoesel and J. Berendsen&quot;,
                &quot;journal&quot;: &quot;Methods and Techniques in Computational Chemistry: METECC-95&quot;,
                &quot;doi&quot;: null,
                &quot;created_at&quot;: &quot;2021-09-30T08:27:11.512Z&quot;,
                &quot;updated_at&quot;: &quot;2021-09-30T08:27:11.512Z&quot;
            }
        ],
        &quot;licence_links&quot;: [],
        &quot;grants&quot;: [
            {
                &quot;id&quot;: 7488,
                &quot;fairsharing_record_id&quot;: 1443,
                &quot;organisation_id&quot;: 230,
                &quot;relation&quot;: &quot;maintains&quot;,
                &quot;created_at&quot;: &quot;2021-09-30T09:28:39.405Z&quot;,
                &quot;updated_at&quot;: &quot;2021-09-30T09:28:39.405Z&quot;,
                &quot;grant_id&quot;: null,
                &quot;is_lead&quot;: true,
                &quot;saved_state&quot;: {
                    &quot;id&quot;: 230,
                    &quot;name&quot;: &quot;BioExcel CoE&quot;,
                    &quot;types&quot;: [
                        &quot;Consortium&quot;
                    ],
                    &quot;is_lead&quot;: true,
                    &quot;relation&quot;: &quot;maintains&quot;
                }
            },
            {
                &quot;id&quot;: 7489,
                &quot;fairsharing_record_id&quot;: 1443,
                &quot;organisation_id&quot;: 913,
                &quot;relation&quot;: &quot;funds&quot;,
                &quot;created_at&quot;: &quot;2021-09-30T09:28:39.443Z&quot;,
                &quot;updated_at&quot;: &quot;2021-09-30T09:32:30.296Z&quot;,
                &quot;grant_id&quot;: 1650,
                &quot;is_lead&quot;: false,
                &quot;saved_state&quot;: {
                    &quot;id&quot;: 913,
                    &quot;name&quot;: &quot;European Commision&quot;,
                    &quot;grant&quot;: &quot;675728&quot;,
                    &quot;types&quot;: [
                        &quot;Undefined&quot;
                    ],
                    &quot;is_lead&quot;: false,
                    &quot;relation&quot;: &quot;funds&quot;
                }
            }
        ],
        &quot;url_for_logo&quot;: null,
        &quot;exhaustive_licences&quot;: false
    }
}
</pre></div>
</div>
</div>
</div>
</section>
<section id="output-the-data-in-a-presentable-format">
<h3>2.3 - Output the data in a presentable format<a class="headerlink" href="#output-the-data-in-a-presentable-format" title="Permalink to this headline">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># here we loop over the data that has been returned and print it out, one per line</span>
<span class="k">for</span> <span class="n">hit</span> <span class="ow">in</span> <span class="n">hits</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">]:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">hit</span><span class="p">[</span><span class="s1">&#39;attributes&#39;</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="s2">&quot;: &quot;</span> <span class="o">+</span> <span class="n">hit</span><span class="p">[</span><span class="s1">&#39;attributes&#39;</span><span class="p">][</span><span class="s1">&#39;url&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>FAIRsharing record for: Portable reduced-precision binary format for trajectories produced by GROMACS package.: https://fairsharing.org/10.25504/FAIRsharing.cb1adb
FAIRsharing record for: Chemistry: https://fairsharing.org/fairsharing_records/3524
FAIRsharing record for: Chemistry vocabulary: https://fairsharing.org/10.25504/FAIRsharing.TrcBD2
FAIRsharing record for: EMODnet Chemistry: https://fairsharing.org/10.25504/FAIRsharing.KOiDmy
FAIRsharing record for: Royal Society of Chemistry - Data policy: https://fairsharing.org/10.25504/FAIRsharing.egbgwm
FAIRsharing record for: MINAS - A Database of Metal Ions in Nucleic AcidS: https://fairsharing.org/10.25504/FAIRsharing.wqtfkv
FAIRsharing record for: ChemSpider: https://fairsharing.org/10.25504/FAIRsharing.96f3gm
FAIRsharing record for: ioChem-BD: https://fairsharing.org/10.25504/FAIRsharing.lwW6a1
FAIRsharing record for: Beilstein Journal of Organic Chemistry: https://fairsharing.org/10.25504/FAIRsharing.7GA79k
FAIRsharing record for: Swedish Ocean Archive: https://fairsharing.org/10.25504/FAIRsharing.npf403
FAIRsharing record for: CHARMM Card File Format: https://fairsharing.org/10.25504/FAIRsharing.7hp91k
FAIRsharing record for: CAS Registry Number: https://fairsharing.org/10.25504/FAIRsharing.r7Kwy7
FAIRsharing record for: Chemical Markup Language: https://fairsharing.org/10.25504/FAIRsharing.3mdt9n
FAIRsharing record for: Data Standard for Sharing Quantitative Results in Mass Spectrometry Metabolomics: https://fairsharing.org/10.25504/FAIRsharing.207caf
FAIRsharing record for: IUPAC International Chemical Identifier: https://fairsharing.org/10.25504/FAIRsharing.ddk9t9
FAIRsharing record for: NFDI4Chem : https://fairsharing.org/fairsharing_records/5027
FAIRsharing record for: Dot Bracket Notation (DBN) - Vienna Format: https://fairsharing.org/10.25504/FAIRsharing.4xrzw1
FAIRsharing record for: Name Reaction Ontology: https://fairsharing.org/10.25504/FAIRsharing.w4tncg
FAIRsharing record for: Molecular Process Ontology: https://fairsharing.org/10.25504/FAIRsharing.mct09a
FAIRsharing record for: Chemical Methods Ontology: https://fairsharing.org/10.25504/FAIRsharing.9j4wh2
FAIRsharing record for: Chemical Abstracts Service Common Chemistry: https://fairsharing.org/10.25504/FAIRsharing.e5c86e
FAIRsharing record for: Geological and Environmental Reference Materials: https://fairsharing.org/10.25504/FAIRsharing.219aec
FAIRsharing record for: MolMeDB: Molecules on Membranes Database: https://fairsharing.org/10.25504/FAIRsharing.CWzk3C
FAIRsharing record for: IUPAC Periodic Table of the Elements: https://fairsharing.org/10.25504/FAIRsharing.92473c
FAIRsharing record for: Network for the Detection of Atmospheric Composition Change: https://fairsharing.org/10.25504/FAIRsharing.298d50
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./manipulations"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="computing_inchis.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Computing InChIs</p>
      </div>
    </a>
    <a class="right-next"
       href="excel_import.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Importing Chemical Data into a Spreadsheet</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#scenario">Scenario</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-searching-pubchem-for-datasets">Step 1 - Searching PubChem for datasets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-the-python-functions">1.1 - Load the Python functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#search-for-sources-that-have-curation-efforts">1.2 - Search for sources that have ‘curation efforts’</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-searching-fair-sharing-for-datasets">Step 2 - Searching FAIR Sharing for datasets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#authentication-to-the-fairsharing-api">2.1 - Authentication to the FAIRSharing API</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#make-the-api-request">2.2 - Make the API request</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#output-the-data-in-a-presentable-format">2.3 - Output the data in a presentable format</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By IUPAC WorldFAIR Chemistry D3.2 Project
</p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
<div class="extra_footer">
  Licenses:  See the "About this..." section in a page for the per resource license. Default license is <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="_blank">CC BY-SA 4.0</a>.<br/>
<em>WorldFAIR “Global cooperation on FAIR data policy and practice” is funded by the EC HORIZON-WIDERA-2021-ERA-01-41<br/>
Coordination and Support Action under Grant Agreement No. 101058393.</em>

</div>
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>