{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Finding a Findable Dataset\n",
    "\n",
    "```{dropdown} About this interactive ![icons](../images/rocket.png) recipe\n",
    "- Author(s): [Stuart Chalk](https://orcid.org/0000-0002-0703-7776)\n",
    "- Reviewer:\n",
    "- Topic(s): How and where to find a 'findable' chemical dataset\n",
    "- Format(s): Interactive Jupyter Notebook (Python)\n",
    "- Scenario(s): You are looking for research data to complement your compare with your own data\n",
    "- Skill(s): You should be familiar with\n",
    "    - [Application Programming Interfaces (APIs)](https://www.ibm.com/topics/api)\n",
    "    - [Working with a data model](https://doi.org/10.1515/pac-2021-3013)\n",
    "    - [Introductory JSON](https://www.youtube.com/watch?v=iiADhChRriM)\n",
    "- Learning outcomes: After completing this example you should understand:\n",
    "    - How to make a request to a website using the Python 'requests' functionality\n",
    "    - Retrieve data in JSON format and how to parse it (knowing the data model)\n",
    "    - How to store confidential data in a remote file\n",
    "    - How programmtically you can authenticate to an API (one of many ways)\n",
    "- Citation: We will not be providing citations for the sampler, but citations for recipes will be available (with copy button) in the full release of the Cookbook and will include DOIâ€™s.\n",
    "- Reuse: This notebook is made available under a [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/) license.\n",
    "```"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Scenario\n",
    "Our group has a set of thermophysical data on over 8000 chemical substances.  We want to integrate into this dataset another physical property dataset so that we can do an analysis of the correlations of the thermophysical data with the chosen physical property of the substances (that are common to both sets).\n",
    "\n",
    "Criteria for picking the physical property dataset: high quality, trusted, large, available with an open license, so I can publish the results and make the derived dataset open.\n",
    "- **High quality means**: unambiguous identification of each chemical substance, enough contextual information (metadata) to make the values scientifically useful, i.e., at least the composition of the solvent, the temperature and for volatile substances the pressure.\n",
    "- **Trusted means**: the provenance chain is reported with the data, and it shows that the data comes from a reputable source(s) and any aggregation and/or processing is documented in enough detail that the community can understand how the dataset has been created/provided."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Step 1 - Searching PubChem for datasets\n",
    "Pubchem houses a lot of data about chemical substances, compounds and bioassays.  Over time external organizations have worked with\n",
    "PubChem to include data, in one of a couple of ways:\n",
    "- data that has been integrated into PubChem pages (e.g., [CCDC](https://pubchem.ncbi.nlm.nih.gov/source/941) -> [example](https://pubchem.ncbi.nlm.nih.gov/compound/241))\n",
    "- data that is not available in a PubChem page but is available via the data sources section of the site as 'annotations' (e.g. [RCSB PDB](https://pubchem.ncbi.nlm.nih.gov/source/15751) -> [Example](https://pubchem.ncbi.nlm.nih.gov/source/15751#data=Annotations))\n",
    "\n",
    "The data available is may not be structured and or clearly described, however if the source has a website with an API then you are likely to get better quality metadata from the linked site."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 1.1 - Load the Python functions"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "outputs": [],
   "source": [
    "# as these are direct imports (they do not reference a Python package) they are built into Python\n",
    "import requests\n",
    "import json"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 1.2 - Search for sources that have 'curation efforts'"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[\n",
      "    {\n",
      "        \"name\": \"Alliance of Genome Resources\",\n",
      "        \"url\": \"https://www.alliancegenome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Barrie Walker, BARK Information Services\",\n",
      "        \"url\": \"https://uk.linkedin.com/in/barrie-walker-85b4a510\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BindingDB\",\n",
      "        \"url\": \"https://www.bindingdb.org/rwd/bind/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioCyc\",\n",
      "        \"url\": \"https://biocyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"BioGRID\",\n",
      "        \"url\": \"https://thebiogrid.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CAMEO Chemicals\",\n",
      "        \"url\": \"https://cameochemicals.noaa.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Catalogue of Life (COL)\",\n",
      "        \"url\": \"https://www.catalogueoflife.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CCSbase\",\n",
      "        \"url\": \"https://ccsbase.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"CDC-ATSDR Toxic Substances Portal\",\n",
      "        \"url\": \"https://www.atsdr.cdc.gov/features/toxicsubstances/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Line Ontology (CLO)\",\n",
      "        \"url\": \"http://www.clo-ontology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Cell Ontology (CL)\",\n",
      "        \"url\": \"http://www.obofoundry.org/ontology/cl.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEBI\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chebi/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChEMBL\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/chembl/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemical Probes Portal\",\n",
      "        \"url\": \"https://www.chemicalprobes.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ChemIDplus\",\n",
      "        \"url\": \"https://pubchem.ncbi.nlm.nih.gov/source/ChemIDplus\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chemoproteomic Metabolic Pathway Resource, Scripps University\",\n",
      "        \"url\": \"https://rc2resource.scripps.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Chris Southan\",\n",
      "        \"url\": \"https://www.ed.ac.uk/medicine-vet-medicine\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Comparative Toxicogenomics Database (CTD)\",\n",
      "        \"url\": \"http://ctdbase.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"COVID-19 Disease Map\",\n",
      "        \"url\": \"https://covid.pages.uni.lu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Gene Interaction database (DGIdb)\",\n",
      "        \"url\": \"http://www.dgidb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Drug Induced Liver Injury Rank (DILIrank) Dataset\",\n",
      "        \"url\": \"https://www.fda.gov/science-research/liver-toxicity-knowledge-base-ltkb/drug-induced-liver-injury-rank-dilirank-dataset\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugBank\",\n",
      "        \"url\": \"https://www.drugbank.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"DrugCentral\",\n",
      "        \"url\": \"http://drugcentral.org\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"E. coli Metabolome Database (ECMDB)\",\n",
      "        \"url\": \"https://ecmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Egon Willighagen (Dept. of Bioinformatics - BiGCaT, Maastricht University)\",\n",
      "        \"url\": \"https://chem-bla-ics.blogspot.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA Air Toxics\",\n",
      "        \"url\": \"https://www3.epa.gov/ttn/atw/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"EPA DSSTox\",\n",
      "        \"url\": \"https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Essentials of Glycobiology, 3rd edition\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK310273/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Orange Book\",\n",
      "        \"url\": \"https://www.fda.gov/drugs/drug-approvals-and-databases/approved-drug-products-therapeutic-equivalence-evaluations-orange-book\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"FDA Pharm Classes\",\n",
      "        \"url\": \"https://www.fda.gov/industry/structured-product-labeling-resources/pharmacologic-class\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gene Ontology (GO)\",\n",
      "        \"url\": \"http://geneontology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Gloriam Group: Computational Receptor Biology - Department of Drug Design and Pharmacology, University of Copenhagen\",\n",
      "        \"url\": \"https://drug.ku.dk/disciplines/peptides-and-proteins/gloriam-group/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlycoNAVI\",\n",
      "        \"url\": \"https://glyconavi.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"GlyGen\",\n",
      "        \"url\": \"https://www.glygen.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Haz-Map, Information on Hazardous Chemicals and Occupational Diseases\",\n",
      "        \"url\": \"https://haz-map.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Chemical Information System (HCIS), Safe Work Australia\",\n",
      "        \"url\": \"http://hcis.safeworkaustralia.gov.au/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Hazardous Substances Data Bank (HSDB)\",\n",
      "        \"url\": \"https://www.nlm.nih.gov/toxnet/index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"HUGO Gene Nomenclature Committee (HGNC)\",\n",
      "        \"url\": \"https://www.genenames.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Human Metabolome Database (HMDB)\",\n",
      "        \"url\": \"https://hmdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"ILO International Chemical Safety Cards (ICSC)\",\n",
      "        \"url\": \"http://www.ilo.org/dyn/icsc/showcard.home\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"INOH\",\n",
      "        \"url\": \"https://www.inoh.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPAC Periodic Table of the Elements and Isotopes (IPTEI)\",\n",
      "        \"url\": \"https://iupac.org/iptei/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"IUPHAR/BPS Guide to PHARMACOLOGY\",\n",
      "        \"url\": \"https://www.guidetopharmacology.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"KEGG\",\n",
      "        \"url\": \"https://www.genome.jp/kegg/kegg2.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"KNApSAcK Species-Metabolite Database\",\n",
      "        \"url\": \"http://www.knapsackfamily.com/KNApSAcK/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Kruve Lab, Ionization & Mass Spectrometry, Stockholm University\",\n",
      "        \"url\": \"https://kruvelab.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Lab and Research Safety, University of Minnesota\",\n",
      "        \"url\": \"https://dehs.umn.edu/department-environmental-health-safety/lab-research-safety\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LIPID MAPS\",\n",
      "        \"url\": \"https://lipidmaps.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"LOTUS - the natural products occurrence database\",\n",
      "        \"url\": \"https://lotus.naturalproducts.net/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"MarkerDB\",\n",
      "        \"url\": \"https://markerdb.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Medical Subject Headings (MeSH)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/mesh\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging and Contrast Agent Database (MICAD)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK5330/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Molecular Imaging Database (MOLI)\",\n",
      "        \"url\": \"https://www.ncbi.nlm.nih.gov/books/NBK5330/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NCI Thesaurus (NCIt)\",\n",
      "        \"url\": \"https://ncit.nci.nih.gov/ncitbrowser/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NEQUIM - Chemoinformatics Group\",\n",
      "        \"url\": \"http://nequim.qui.ufmg.br/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NextMove Software\",\n",
      "        \"url\": \"https://www.nextmovesoftware.com/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIAID ChemDB\",\n",
      "        \"url\": \"https://chemdb.niaid.nih.gov/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NIST Chemistry WebBook\",\n",
      "        \"url\": \"https://webbook.nist.gov/chemistry/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"NITE-CMC\",\n",
      "        \"url\": \"https://www.nite.go.jp/chem/english/ghs/ghs_index.html\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Online Mendelian Inheritance in Man (OMIM)\",\n",
      "        \"url\": \"https://omim.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PANTHER\",\n",
      "        \"url\": \"http://www.pantherdb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PathBank\",\n",
      "        \"url\": \"https://pathbank.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pathway Interaction Database\",\n",
      "        \"url\": \"https://wiki.nci.nih.gov/pages/viewpage.action?pageId=315491760\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pfam\",\n",
      "        \"url\": \"https://www.ebi.ac.uk/interpro/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PharmGKB\",\n",
      "        \"url\": \"https://www.pharmgkb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Pistoia Alliance DataFAIRy Bioassay Pilot\",\n",
      "        \"url\": \"https://www.pistoiaalliance.org/projects/current-projects/datafairy-project/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Plant Reactome\",\n",
      "        \"url\": \"https://plantreactome.gramene.org/index.php?lang=en\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"PlantCyc\",\n",
      "        \"url\": \"https://www.plantcyc.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Protein Ontology\",\n",
      "        \"url\": \"https://proconsortium.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"RCSB Protein Data Bank (RCSB PDB)\",\n",
      "        \"url\": \"https://www.rcsb.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Reactome\",\n",
      "        \"url\": \"https://reactome.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Rhea - Annotated Reactions Database\",\n",
      "        \"url\": \"https://www.rhea-db.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"SureChEMBL\",\n",
      "        \"url\": \"https://www.surechembl.org/search/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics Cellosaurus\",\n",
      "        \"url\": \"https://www.cellosaurus.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Swiss Institute of Bioinformatics ENZYME\",\n",
      "        \"url\": \"https://enzyme.expasy.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Cambridge Structural Database\",\n",
      "        \"url\": \"https://www.ccdc.cam.ac.uk/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The National Institute for Occupational Safety and Health (NIOSH)\",\n",
      "        \"url\": \"https://www.cdc.gov/niosh/npg/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The Natural Products Atlas\",\n",
      "        \"url\": \"https://www.npatlas.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"The University of Alabama Libraries\",\n",
      "        \"url\": \"https://ir.ua.edu/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Therapeutic Target Database (TTD)\",\n",
      "        \"url\": \"https://db.idrblab.net/ttd/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Toxin and Toxin Target Database (T3DB)\",\n",
      "        \"url\": \"http://www.t3db.ca/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniCarbKB\",\n",
      "        \"url\": \"http://www.unicarbkb.org\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"UniProt\",\n",
      "        \"url\": \"https://www.uniprot.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikidata\",\n",
      "        \"url\": \"https://www.wikidata.org/wiki/Wikidata:Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Wikipedia\",\n",
      "        \"url\": \"https://en.wikipedia.org/wiki/Main_Page\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"World Register of Marine Species (WoRMS)\",\n",
      "        \"url\": \"https://www.marinespecies.org/\"\n",
      "    },\n",
      "    {\n",
      "        \"name\": \"Yeast Metabolome Database (YMDB)\",\n",
      "        \"url\": \"http://www.ymdb.ca/\"\n",
      "    }\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "# This URL is the metadata about the data sources in PubChem\n",
    "url = 'https://pubchem.ncbi.nlm.nih.gov/rest/pug/sourcetable/all/JSON/?response_type=display'\n",
    "response = requests.get(url)\n",
    "srcs = response.json()\n",
    "results = []\n",
    "search = 'Curation Efforts'  # i.e., a repository, or other type of data source (this is in index 8 of the data list for each source)\n",
    "rows = srcs['Table']['Row']\n",
    "for row in rows:\n",
    "    if row['Cell'][8].find(search) != -1:\n",
    "        hit = {}\n",
    "        hit.update({'name': row['Cell'][0]})\n",
    "        hit.update({'url': row['Cell'][9]})\n",
    "        results.append(hit)\n",
    "# when printed this is a scrollable list of many entries\n",
    "print(json.dumps(results, indent=4))"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Step 2 - Searching FAIR Sharing for datasets\n",
    "FAIRSharing is a database of FAIR resources and per se a database of datasets, however you might find a repository here that\n",
    "has the kind of data you are looking for. The code below accesses the FAIR sharing API so search for 'chemistry' (or other term)\n",
    "related resources."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.1 - Authentication to the FAIRSharing API"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'success': True, 'jwt': 'eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiIxMzBmOGNjYS1jNzY3LTRkMGQtYjZmMC04ODIwODEwNGIzNzYiLCJzdWIiOiI4NTA5Iiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNjc1MTg0NjQ2LCJleHAiOjE2NzUyNzEwNDZ9._bWHFm0KUPi7bKkb-qymgzhSzhr00whlgV6zq9ys-CY', 'username': 'ChemCookbook', 'id': 8509, 'role': 'user', 'profile_type': 'none', 'watched_records': [], 'is_curator': False, 'is_super_curator': False, 'third_party': False, 'expiry': 1675271046, 'message': 'Authentication successful'}\n"
     ]
    }
   ],
   "source": [
    "# see https://fairsharing.org/API_doc for instructions on how to search the API\n",
    "from hiddensettings import *\n",
    "# user login\n",
    "url = 'https://api.fairsharing.org/users/sign_in'\n",
    "loghdrs = {'Accept': 'application/json','Content-Type': 'application/json'}\n",
    "login = {'user': {'login': fs_user, 'password': fs_pass}}\n",
    "response = requests.request(\"POST\", url, headers=loghdrs, data=json.dumps(login))\n",
    "data = response.json()\n",
    "print(data)"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.2 - Make the API request"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"id\": \"3524\",\n",
      "    \"type\": \"fairsharing_records\",\n",
      "    \"attributes\": {\n",
      "        \"created_at\": \"2018-07-11T19:43:37.000Z\",\n",
      "        \"updated_at\": \"2022-11-28T14:46:52.777Z\",\n",
      "        \"metadata\": {\n",
      "            \"name\": \"Chemistry\",\n",
      "            \"status\": \"ready\",\n",
      "            \"contacts\": [],\n",
      "            \"homepage\": \"https://www.go-fair.org/implementation-networks/overview/chemistryin/\",\n",
      "            \"citations\": [],\n",
      "            \"identifier\": 3524,\n",
      "            \"description\": \"A curated collection of standards, databases and policies covering all aspects of chemistry, providing a curated view of FAIRsharing chemistry resources. Although this collection was initiated within the GO FAIR Implementation Network for Chemistry (and in tandem with the Chemistry Research Data Interest Group (CRDIG) of the Research Data Alliance), it has grown beyond this project. Any domain experts who would like to contribute to this collection, within the RDA or beyond, are welcome to claim and help maintain it.\",\n",
      "            \"abbreviation\": \"Chemistry\",\n",
      "            \"reference_url\": \"https://www.go-fair.org/implementation-networks/overview/implementation-networks-archive/chemistryin/\",\n",
      "            \"deprecation_reason\": null\n",
      "        },\n",
      "        \"legacy_ids\": [\n",
      "            \"bsg-c000049\"\n",
      "        ],\n",
      "        \"name\": \"FAIRsharing record for: Chemistry\",\n",
      "        \"abbreviation\": \"Chemistry\",\n",
      "        \"url\": \"https://fairsharing.org/fairsharing_records/3524\",\n",
      "        \"doi\": null,\n",
      "        \"fairsharing_licence\": \"https://creativecommons.org/licenses/by-sa/4.0/. Please link to https://fairsharing.org and https://api.fairsharing.org/img/fairsharing-attribution.svg for attribution.\",\n",
      "        \"description\": \"This FAIRsharing record describes: A curated collection of standards, databases and policies covering all aspects of chemistry, providing a curated view of FAIRsharing chemistry resources. Although this collection was initiated within the GO FAIR Implementation Network for Chemistry (and in tandem with the Chemistry Research Data Interest Group (CRDIG) of the Research Data Alliance), it has grown beyond this project. Any domain experts who would like to contribute to this collection, within the RDA or beyond, are welcome to claim and help maintain it.\",\n",
      "        \"linked_records\": [\n",
      "            {\n",
      "                \"linked_record_name\": \"Nuclear Magnetic Resonance Controlled Vocabulary\",\n",
      "                \"linked_record_id\": 1021,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11706\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Chemical Entities of Biological Interest\",\n",
      "                \"linked_record_id\": 1036,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11707\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CHEMical INFormation Ontology\",\n",
      "                \"linked_record_id\": 1056,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11708\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Suggested Ontology for PHARMacogenomics\",\n",
      "                \"linked_record_id\": 55,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11709\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Standards for Reporting Enzymology Data Guidelines\",\n",
      "                \"linked_record_id\": 1292,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11710\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CDISC Laboratory Data Model\",\n",
      "                \"linked_record_id\": 1507,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11711\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Core Information for Metabolomics Reporting\",\n",
      "                \"linked_record_id\": 665,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11712\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Chemical Markup Language\",\n",
      "                \"linked_record_id\": 91,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11713\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"MDL molfile Format\",\n",
      "                \"linked_record_id\": 1480,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11714\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Minimal Information Required In the Annotation of Models\",\n",
      "                \"linked_record_id\": 988,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11715\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CHARMM Card File Format\",\n",
      "                \"linked_record_id\": 118,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11716\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Minimum Information about a Molecular Interaction Experiment\",\n",
      "                \"linked_record_id\": 1279,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11717\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Crystallographic Information Framework\",\n",
      "                \"linked_record_id\": 183,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11718\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Core Scientific MetaData model\",\n",
      "                \"linked_record_id\": 391,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11719\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"National Drug File\",\n",
      "                \"linked_record_id\": 155,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11720\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"National Drug Data File\",\n",
      "                \"linked_record_id\": 1073,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11721\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Pharmacogenomic Relationships Ontology\",\n",
      "                \"linked_record_id\": 215,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11722\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Cancer Chemoprevention Ontology\",\n",
      "                \"linked_record_id\": 898,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11723\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Master Drug Data Base Clinical Drugs\",\n",
      "                \"linked_record_id\": 680,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11724\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Toxicology Data Markup Language\",\n",
      "                \"linked_record_id\": 374,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11725\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Minimum Information Required for a Drug Metabolism Enzymes and Transporters Experiment\",\n",
      "                \"linked_record_id\": 1192,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11726\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Analytical Information Markup Language\",\n",
      "                \"linked_record_id\": 1172,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11727\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Joint Committee on Atomic and Molecular Physical data - working group on Data eXchange\",\n",
      "                \"linked_record_id\": 111,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11728\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Nuclear Magnetic Resonance Markup Language\",\n",
      "                \"linked_record_id\": 1363,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11729\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Pharmacometrics Markup Language\",\n",
      "                \"linked_record_id\": 16,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11730\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"IUPAC International Chemical Identifier\",\n",
      "                \"linked_record_id\": 1477,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"identifier_schema\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11731\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"eNanoMapper Ontology\",\n",
      "                \"linked_record_id\": 493,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11732\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"ChemDraw Native File Format\",\n",
      "                \"linked_record_id\": 1343,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11733\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Minimum Information About a Simulation Experiment\",\n",
      "                \"linked_record_id\": 1260,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"reporting_guideline\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11734\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Structure Data Format\",\n",
      "                \"linked_record_id\": 1462,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11735\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Drug Target Ontology\",\n",
      "                \"linked_record_id\": 1469,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11736\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"ThermoML\",\n",
      "                \"linked_record_id\": 1091,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11737\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"AnaEE Thesaurus\",\n",
      "                \"linked_record_id\": 1545,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11738\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Hierarchical Editing Language for Macromolecules\",\n",
      "                \"linked_record_id\": 1108,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"model_and_format\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11739\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CAS Registry Number\",\n",
      "                \"linked_record_id\": 381,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"identifier_schema\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11740\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Chemistry vocabulary\",\n",
      "                \"linked_record_id\": 101,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11741\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"IUPAC Compendium of Chemical Terminology\",\n",
      "                \"linked_record_id\": 46,\n",
      "                \"linked_record_registry\": \"Standard\",\n",
      "                \"linked_record_type\": \"terminology_artefact\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11742\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Springer Nature Research Data Policy Type 1\",\n",
      "                \"linked_record_id\": 3360,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal_publisher\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11743\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Springer Nature Research Data Policy Type 4\",\n",
      "                \"linked_record_id\": 3415,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal_publisher\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11744\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"American Chemical Society - Accounts of Chemical Research  - Information for Authors\",\n",
      "                \"linked_record_id\": 3400,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11745\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"American Chemical Society - Chemical Reviews - Information for Authors\",\n",
      "                \"linked_record_id\": 3472,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11746\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Springer Nature Research Data Policy Type 2\",\n",
      "                \"linked_record_id\": 3425,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal_publisher\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11747\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Hindawi Research Data Policy\",\n",
      "                \"linked_record_id\": 3412,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11748\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Nature Chemical Biology - Reporting standards and availability of data, materials, code and protocols\",\n",
      "                \"linked_record_id\": 3486,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11749\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Springer Nature Research Data Policy Type 3 (for life sciences)\",\n",
      "                \"linked_record_id\": 3416,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal_publisher\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11750\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Royal Society of Chemistry - Data policy\",\n",
      "                \"linked_record_id\": 3410,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11751\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Nature Chemistry - Reporting standards and availability of data, materials, code and protocols\",\n",
      "                \"linked_record_id\": 3371,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11752\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Springer Nature Research Data Policy Type 3\",\n",
      "                \"linked_record_id\": 3426,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal_publisher\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11753\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"The Plant Cell - Instructions for Authors\",\n",
      "                \"linked_record_id\": 3446,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11754\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Cell Press - Trends In Biochemical Sciences - Instructions for Authors\",\n",
      "                \"linked_record_id\": 3361,\n",
      "                \"linked_record_registry\": \"Policy\",\n",
      "                \"linked_record_type\": \"journal\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11755\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"GlycoNAVI\",\n",
      "                \"linked_record_id\": 1548,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11756\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"canSAR\",\n",
      "                \"linked_record_id\": 1559,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11757\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Crystallography Open Database\",\n",
      "                \"linked_record_id\": 1562,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11758\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"MINAS - A Database of Metal Ions in Nucleic AcidS\",\n",
      "                \"linked_record_id\": 1607,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11759\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Polbase\",\n",
      "                \"linked_record_id\": 1632,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase_and_repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11760\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Protein-Chemical Structural Interactions\",\n",
      "                \"linked_record_id\": 1636,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11761\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Rhea\",\n",
      "                \"linked_record_id\": 1639,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11762\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Syntheses, Chemicals, and Reactions In Patents DataBase\",\n",
      "                \"linked_record_id\": 1642,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11763\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Search Tool for Interactions of Chemicals\",\n",
      "                \"linked_record_id\": 1648,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase_and_repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11764\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"SuperTarget\",\n",
      "                \"linked_record_id\": 1691,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11765\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"TDR Targets\",\n",
      "                \"linked_record_id\": 1692,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11766\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Comparative Toxicogenomics Database\",\n",
      "                \"linked_record_id\": 1716,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11767\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Chemical Abstracts Service Registry\",\n",
      "                \"linked_record_id\": 1747,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11768\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"The Cambridge Structural Database\",\n",
      "                \"linked_record_id\": 1796,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11769\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Chemical Component Dictionary\",\n",
      "                \"linked_record_id\": 1847,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11770\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Small Molecule Pathway Database\",\n",
      "                \"linked_record_id\": 1887,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase_and_repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11771\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"SABIO-RK Biochemical Reaction Kinetics Database\",\n",
      "                \"linked_record_id\": 1891,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11772\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"J-GLOBAL\",\n",
      "                \"linked_record_id\": 1936,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11773\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Three-Dimensional Structure Database of Natural Metabolites\",\n",
      "                \"linked_record_id\": 1999,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11774\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"ChemIDplus\",\n",
      "                \"linked_record_id\": 2003,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11775\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"ChemSpider\",\n",
      "                \"linked_record_id\": 2042,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11776\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"The Human Metabolome Database\",\n",
      "                \"linked_record_id\": 2084,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11777\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"The UC Irvine ChemDB\",\n",
      "                \"linked_record_id\": 2088,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11778\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"eCrystals\",\n",
      "                \"linked_record_id\": 2108,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11779\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"SigMol\",\n",
      "                \"linked_record_id\": 2241,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase_and_repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11780\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"MetaNetX\",\n",
      "                \"linked_record_id\": 2281,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11781\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"data.eNanoMapper.net\",\n",
      "                \"linked_record_id\": 2282,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11782\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Wikidata\",\n",
      "                \"linked_record_id\": 2329,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11783\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Open Targets\",\n",
      "                \"linked_record_id\": 2369,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11784\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"SureChEMBL\",\n",
      "                \"linked_record_id\": 2378,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11785\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"EarthChem\",\n",
      "                \"linked_record_id\": 2424,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11786\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"EPA Comptox Chemicals Dashboard\",\n",
      "                \"linked_record_id\": 2480,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11787\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"High Energy Physics Data Repository\",\n",
      "                \"linked_record_id\": 2497,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11788\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CSIRO Data Access Portal\",\n",
      "                \"linked_record_id\": 2508,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11789\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"FlavorDB\",\n",
      "                \"linked_record_id\": 2516,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11790\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Target Pathogen\",\n",
      "                \"linked_record_id\": 2522,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11791\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"CloudFlame\",\n",
      "                \"linked_record_id\": 2531,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11792\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"4TU.ResearchData\",\n",
      "                \"linked_record_id\": 2537,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11793\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Citrination\",\n",
      "                \"linked_record_id\": 2538,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11794\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"ThermoML Archive\",\n",
      "                \"linked_record_id\": 2545,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11795\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"SuperDRUG2 - A One Stop Resource for Approved/Marketed Drugs\",\n",
      "                \"linked_record_id\": 2590,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"knowledgebase\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11796\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"NASA Ames PAH IR Spectroscopic Database\",\n",
      "                \"linked_record_id\": 2619,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11797\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"NIST Atomic Spectra Database\",\n",
      "                \"linked_record_id\": 2621,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11798\n",
      "            },\n",
      "            {\n",
      "                \"linked_record_name\": \"Kinetic Database for Astrochemistry\",\n",
      "                \"linked_record_id\": 2622,\n",
      "                \"linked_record_registry\": \"Database\",\n",
      "                \"linked_record_type\": \"repository\",\n",
      "                \"relation\": \"collects\",\n",
      "                \"link_id\": 11799\n",
      "            }\n",
      "        ],\n",
      "        \"linking_records\": [],\n",
      "        \"fairsharing_registry\": \"Collection\",\n",
      "        \"record_type\": \"collection\",\n",
      "        \"subjects\": [\n",
      "            \"Chemistry\"\n",
      "        ],\n",
      "        \"domains\": [],\n",
      "        \"taxonomies\": [\n",
      "            \"All\"\n",
      "        ],\n",
      "        \"user_defined_tags\": [],\n",
      "        \"countries\": [\n",
      "            \"Worldwide\"\n",
      "        ],\n",
      "        \"publications\": [],\n",
      "        \"licence_links\": [],\n",
      "        \"grants\": [\n",
      "            {\n",
      "                \"id\": 10157,\n",
      "                \"fairsharing_record_id\": 3524,\n",
      "                \"organisation_id\": 3794,\n",
      "                \"relation\": \"associated_with\",\n",
      "                \"created_at\": \"2022-11-28T13:52:29.395Z\",\n",
      "                \"updated_at\": \"2022-11-28T13:52:29.395Z\",\n",
      "                \"grant_id\": null,\n",
      "                \"is_lead\": false,\n",
      "                \"saved_state\": {\n",
      "                    \"id\": 3794,\n",
      "                    \"name\": \"RDA (Research Data Alliance) Chemistry Research Data Interest Group (CRDIG)\",\n",
      "                    \"types\": [\n",
      "                        \"Consortium\"\n",
      "                    ],\n",
      "                    \"is_lead\": false,\n",
      "                    \"relation\": \"associated_with\"\n",
      "                }\n",
      "            },\n",
      "            {\n",
      "                \"id\": 10158,\n",
      "                \"fairsharing_record_id\": 3524,\n",
      "                \"organisation_id\": 1187,\n",
      "                \"relation\": \"associated_with\",\n",
      "                \"created_at\": \"2022-11-28T13:52:29.442Z\",\n",
      "                \"updated_at\": \"2022-11-28T13:52:29.442Z\",\n",
      "                \"grant_id\": null,\n",
      "                \"is_lead\": false,\n",
      "                \"saved_state\": {\n",
      "                    \"id\": 1187,\n",
      "                    \"name\": \"GO FAIR\",\n",
      "                    \"types\": [\n",
      "                        \"Consortium\"\n",
      "                    ],\n",
      "                    \"is_lead\": false,\n",
      "                    \"relation\": \"associated_with\"\n",
      "                }\n",
      "            }\n",
      "        ],\n",
      "        \"url_for_logo\": null,\n",
      "        \"exhaustive_licences\": false\n",
      "    }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# in order to authenticate when making an API request the 'jwt' code above must\n",
    "# be included in the JSON headers (see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields)\n",
    "jwt = data['jwt']\n",
    "srchdrs = {'Accept': 'application/json', 'Content-Type': 'application/json', 'Authorization': \"Bearer {0}\".format(jwt)}\n",
    "searchterm = 'chemistry'\n",
    "searchurl ='https://api.fairsharing.org/search/fairsharing_records?q=' + searchterm\n",
    "search = requests.request(\"POST\", searchurl, headers=srchdrs)\n",
    "hits = json.loads(search.content)\n",
    "# this prints out the raw JSON for the first entry (the 'data' entry is a JSON list)\n",
    "# that is returned from the API request (formatted nicely, which means its on many lines)\n",
    "print(json.dumps(hits['data'][0], indent=4))"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "### 2.3 - Output the data in a presentable format"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FAIRsharing record for: Chemistry: https://fairsharing.org/fairsharing_records/3524\n",
      "FAIRsharing record for: Portable reduced-precision binary format for trajectories produced by GROMACS package.: https://fairsharing.org/fairsharing_records/1443\n",
      "FAIRsharing record for: Chemistry vocabulary: https://fairsharing.org/10.25504/FAIRsharing.TrcBD2\n",
      "FAIRsharing record for: EMODnet Chemistry: https://fairsharing.org/10.25504/FAIRsharing.KOiDmy\n",
      "FAIRsharing record for: Royal Society of Chemistry - Data policy: https://fairsharing.org/10.25504/FAIRsharing.egbgwm\n",
      "FAIRsharing record for: Beilstein Journal of Organic Chemistry: https://fairsharing.org/10.25504/FAIRsharing.7GA79k\n",
      "FAIRsharing record for: ioChem-BD: https://fairsharing.org/10.25504/FAIRsharing.lwW6a1\n",
      "FAIRsharing record for: MINAS - A Database of Metal Ions in Nucleic AcidS: https://fairsharing.org/10.25504/FAIRsharing.wqtfkv\n",
      "FAIRsharing record for: ChemSpider: https://fairsharing.org/10.25504/FAIRsharing.96f3gm\n",
      "FAIRsharing record for: Chemical Markup Language: https://fairsharing.org/10.25504/FAIRsharing.3mdt9n\n",
      "FAIRsharing record for: mzTab for Metabolomics: https://fairsharing.org/fairsharing_records/831\n",
      "FAIRsharing record for: CAS Registry Number: https://fairsharing.org/10.25504/FAIRsharing.r7Kwy7\n",
      "FAIRsharing record for: IUPAC International Chemical Identifier: https://fairsharing.org/10.25504/FAIRsharing.ddk9t9\n",
      "FAIRsharing record for: CHARMM Card File Format: https://fairsharing.org/10.25504/FAIRsharing.7hp91k\n",
      "FAIRsharing record for: MolMeDB: Molecules on Membranes Database: https://fairsharing.org/10.25504/FAIRsharing.CWzk3C\n",
      "FAIRsharing record for: Swedish Ocean Archive: https://fairsharing.org/10.25504/FAIRsharing.npf403\n",
      "FAIRsharing record for: Chemical Methods Ontology: https://fairsharing.org/10.25504/FAIRsharing.9j4wh2\n",
      "FAIRsharing record for: Name Reaction Ontology: https://fairsharing.org/10.25504/FAIRsharing.w4tncg\n",
      "FAIRsharing record for: Dot Bracket Notation (DBN) - Vienna Format: https://fairsharing.org/10.25504/FAIRsharing.4xrzw1\n",
      "FAIRsharing record for: Molecular Process Ontology: https://fairsharing.org/10.25504/FAIRsharing.mct09a\n",
      "FAIRsharing record for: IUPAC Compendium of Chemical Terminology: https://fairsharing.org/10.25504/FAIRsharing.OsiqlF\n",
      "FAIRsharing record for: Analytical Information Markup Language: https://fairsharing.org/10.25504/FAIRsharing.6cs4bf\n",
      "FAIRsharing record for: Geological and Environmental Reference Materials: https://fairsharing.org/10.25504/FAIRsharing.219aec\n",
      "FAIRsharing record for: SupraBank: https://fairsharing.org/10.25504/FAIRsharing.vjWUT7\n",
      "FAIRsharing record for: Network for the Detection of Atmospheric Composition Change: https://fairsharing.org/10.25504/FAIRsharing.298d50\n"
     ]
    }
   ],
   "source": [
    "# here we loop over the data that has been returned and print it out, one per line\n",
    "for hit in hits['data']:\n",
    "    print(hit['attributes']['name'] + \": \" + hit['attributes']['url'])"
   ],
   "metadata": {
    "collapsed": false
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
